Index: kresources/factory.cpp
===================================================================
--- kresources/factory.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kresources/factory.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -57,9 +57,15 @@
     QMap<QString, KService::Ptr> mTypeMap;
 };
 
-typedef QMap<QString, Factory*> factoryMap;
-K_GLOBAL_STATIC( factoryMap, mSelves )
+class FactoryMap : public QMap<QString, Factory*>
+{
+public:
 
+    ~FactoryMap() { qDeleteAll(*this); }
+};
+
+K_GLOBAL_STATIC( FactoryMap, mSelves )
+
 Factory *Factory::self( const QString &resourceFamily )
 {
   kDebug();
Index: kcal/localdir.desktop
===================================================================
--- kcal/localdir.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/localdir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -41,6 +41,7 @@
 Name[zh_TW]=本地端目錄裡的行事曆
 Comment=Provides access to calendar items, each stored in a single file, in a given directory
 Comment[ca]=Proporciona l'accés als elements d'un calendari, cada un emmagatzemat en un fitxer individual, en un directori donat
+Comment[cs]=Poskytuje přístup k položkám kalendáře, každé uložené v jednom souboru v daném adresáři
 Comment[da]=Giver adgang til kalenderelementer, hver lagret i en enkelt fil, i en given mappe
 Comment[de]=Ermöglicht Zugriff auf Kalender, die jeweils in einzelnen Dateien in einem vorgegebenen Ordner gespeichert sind. 
 Comment[el]=Προσφέρει πρόσβαση σε αντικείμενα ημερολογίου, αποθηκευμένα σε ξεχωριστά αρχεία, σε ένα δοσμένο κατάλογο
Index: kcal/local.desktop
===================================================================
--- kcal/local.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/local.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -41,6 +41,7 @@
 Name[zh_TW]=本地端檔案裡的行事曆
 Comment=Provides access to a calendar stored in a single local file
 Comment[ca]=Proporciona l'accés a un calendari emmagatzemat en un fitxer local individual
+Comment[cs]=Poskytuje přístup ke kalendáři uloženém v jednom lokálním souboru
 Comment[da]=Giver adgang til en kalender lagret i en enkelt lokal fil
 Comment[de]=Ermöglicht Zugriff auf einen Kalender, der in einer einzigen Dateien lokal gespeichert ist
 Comment[el]=Προσφέρει πρόσβαση σε ένα ημερολόγιο αποθηκευμένο σε ένα τοπικό αρχείο
Index: kcal/kcal_manager.desktop
===================================================================
--- kcal/kcal_manager.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/kcal_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -19,6 +19,7 @@
 Name[km]=ប្រតិទិន
 Name[lt]=Kalendorius
 Name[lv]=Kalendārs
+Name[mai]=कैलेंडर
 Name[nb]=Kalender
 Name[nds]=Kalenner
 Name[nl]=Agenda
Index: kcal/recurrence.cpp
===================================================================
--- kcal/recurrence.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/recurrence.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -941,12 +941,35 @@
   for ( i = 0, count = d->mRRules.count();  i < count;  ++i ) {
     times += d->mRRules[i]->timesInInterval( start, end );
   }
-  times += d->mRDateTimes;
-  KDateTime kdt( startDateTime() );
+
+  // add rdatetimes that fit in the interval
+  for ( i = 0, count = d->mRDateTimes.count();  i < count;  ++i ) {
+    if ( d->mRDateTimes[i] >= start && d->mRDateTimes[i] <= end ) {
+      times += d->mRDateTimes[i];
+    }
+  }
+
+  // add rdates that fit in the interval
+  KDateTime kdt( d->mStartDateTime );
   for ( i = 0, count = d->mRDates.count();  i < count;  ++i ) {
     kdt.setDate( d->mRDates[i] );
-    times += kdt;
+    if ( kdt >= start && kdt <= end ) {
+      times += kdt;
+    }
   }
+
+  // Recurrence::timesInInterval(...) doesn't explicitly add mStartDateTime to the list
+  // of times to be returned. It calls mRRules[i]->timesInInterval(...) which include
+  // mStartDateTime.
+  // So, If we have rdates/rdatetimes but don't have any rrule we must explicitly
+  // add mStartDateTime to the list, otherwise we won't see the first occurrence.
+  if ( ( !d->mRDates.isEmpty() || !d->mRDateTimes.isEmpty() ) &&
+       d->mRRules.isEmpty() &&
+       start <= d->mStartDateTime &&
+       end >= d->mStartDateTime ) {
+    times += d->mStartDateTime;
+  }
+
   times.sortUnique();
 
   // Remove excluded times
Index: kcal/htmlexport.cpp
===================================================================
--- kcal/htmlexport.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/htmlexport.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -501,7 +501,7 @@
     *ts << " class=\"done\"";
   }
   *ts << ">\n";
-  *ts << "    " << i18nc( "@info to-do percent complete",
+  *ts << "    " << i18nc( "@info/plain to-do percent complete",
                           "%1 %", todo->percentComplete() ) << "\n";
   *ts << "  </td>\n";
 
@@ -664,7 +664,7 @@
 void HtmlExport::createFooter( QTextStream *ts )
 {
   // FIXME: Implement this in a translatable way!
-  QString trailer = i18nc( "@info", "This page was created " );
+  QString trailer = i18nc( "@info/plain", "This page was created " );
 
 /*  bool hasPerson = false;
   bool hasCredit = false;
@@ -672,27 +672,27 @@
   QString mail, name, credit, creditURL;*/
   if ( !d->mSettings->eMail().isEmpty() ) {
     if ( !d->mSettings->name().isEmpty() ) {
-      trailer += i18nc( "@info page creator email link with name",
+      trailer += i18nc( "@info/plain page creator email link with name",
                         "by <link url='mailto:%1'>%2</link>",
                         d->mSettings->eMail(), d->mSettings->name() );
     } else {
-      trailer += i18nc( "@info page creator email link",
+      trailer += i18nc( "@info/plain page creator email link",
                         "by <link url='mailto:%1'>%2</link>",
                         d->mSettings->eMail(), d->mSettings->eMail() );
     }
   } else {
     if ( !d->mSettings->name().isEmpty() ) {
-      trailer += i18nc( "@info page creator name only",
+      trailer += i18nc( "@info/plain page creator name only",
                         "by %1 ", d->mSettings->name() );
     }
   }
   if ( !d->mSettings->creditName().isEmpty() ) {
     if ( !d->mSettings->creditURL().isEmpty() ) {
-      trailer += i18nc( "@info page credit with name and link",
+      trailer += i18nc( "@info/plain page credit with name and link",
                         "with <link url='%1'>%2</link>",
                         d->mSettings->creditURL(), d->mSettings->creditName() );
     } else {
-      trailer += i18nc( "@info page credit name only",
+      trailer += i18nc( "@info/plain page credit name only",
                         "with %1", d->mSettings->creditName() );
     }
   }
@@ -759,7 +759,7 @@
   if ( d->mHolidayMap[date].isEmpty() ) {
     d->mHolidayMap[date] = name;
   } else {
-    d->mHolidayMap[date] = i18nc( "@info holiday by date and name",
+    d->mHolidayMap[date] = i18nc( "@info/plain holiday by date and name",
                                   "%1, %2", d->mHolidayMap[date], name );
   }
 }
Index: kcal/incidenceformatter.cpp
===================================================================
--- kcal/incidenceformatter.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/incidenceformatter.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -402,11 +402,12 @@
     tmpStr += "<td>" + eventViewerFormatAttachments( event ) + "</td>";
     tmpStr += "</tr>";
   }
-
+  KDateTime kdt = event->created().toTimeSpec( spec );
   tmpStr += "</table>";
   tmpStr += "<p><em>" +
             i18n( "Creation date: %1", KGlobal::locale()->formatDateTime(
-                    event->created().dateTime(), KLocale::ShortDate ) ) + "</em>";
+                    kdt.dateTime(),
+                    KLocale::ShortDate ) ) + "</em>";
   return tmpStr;
 }
 
@@ -449,8 +450,10 @@
   }
   tmpStr += eventViewerFormatAttendees( todo );
   tmpStr += eventViewerFormatAttachments( todo );
+
+  KDateTime kdt = todo->created().toTimeSpec( spec );
   tmpStr += "<p><em>" + i18n( "Creation date: %1",
-    KGlobal::locale()->formatDateTime( todo->created().dateTime(), KLocale::ShortDate ) ) + "</em>";
+    KGlobal::locale()->formatDateTime( kdt.dateTime(), KLocale::ShortDate ) ) + "</em>";
   return tmpStr;
 }
 
@@ -655,26 +658,41 @@
 {
   QString html;
   QString descr;
-  if ( !incidence->descriptionIsRich() ) {
-    descr = string2HTML( incidence->description() );
-  } else {
-    descr = incidence->richDescription();
-    if ( noHtmlMode ) {
-      descr = cleanHtml( descr );
+  QStringList comments;
+  if ( !incidence->description().isEmpty() ) {
+    if ( !incidence->descriptionIsRich() ) {
+      descr = string2HTML( incidence->description() );
+    } else {
+      descr = incidence->richDescription();
+      if ( noHtmlMode ) {
+        descr = cleanHtml( descr );
+      }
+      descr = eventViewerAddTag( "p", descr );
     }
-    descr = eventViewerAddTag( "p", descr );
   }
+
+  if ( incidence->comments().isEmpty() && !descr.isEmpty() ) {
+    comments << descr;
+  } else {
+    comments = incidence->comments();
+  }
+
   if( !descr.isEmpty() ) {
     html += "<br/><u>" + i18n( "Description:" ) + "</u><table border=\"0\"><tr><td>&nbsp;</td><td>";
     html += descr + "</td></tr></table>";
   }
-  QStringList comments = incidence->comments();
   if ( !comments.isEmpty() ) {
-    html += "<br><u>" + i18n( "Comments:" ) + "</u><table border=\"0\"><tr><td>&nbsp;</td><td><ul>";
-    for ( int i = 0; i < comments.count(); ++i ) {
-      html += "<li>" + string2HTML( comments[i] ) + "</li>";
+    html += "<br><u>" + i18n( "Comments:" ) + "</u><table border=\"0\"><tr><td>&nbsp;</td><td>";
+    if ( comments.count() > 1 ) {
+      html += "<ul>";
+      for ( int i = 0; i < comments.count(); ++i ) {
+        html += "<li>" + string2HTML( comments[i] ) + "</li>";
+      }
+      html += "</ul>";
+    } else {
+      html += string2HTML( comments[0] );
     }
-    html += "</ul></td></tr></table>";
+    html += "</td></tr></table>";
   }
   return html;
 }
Index: kcal/calendar.cpp
===================================================================
--- kcal/calendar.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/calendar.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -457,7 +457,7 @@
 
   Incidence *newInc = incidence->clone();
   newInc->recreate();
-  newInc->setRelatedTo( incidence );
+  //newInc->setRelatedTo( incidence );
   Recurrence *recur = newInc->recurrence();
   if ( single ) {
     recur->clear();
Index: kcal/recurrencerule.cpp
===================================================================
--- kcal/recurrencerule.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kcal/recurrencerule.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -751,7 +751,6 @@
                                // unless it matches the rule)
     uint mFrequency;
     /** how often it recurs:
-          -1 means infinitely,
            0 means an explicit end date,
            positive values give the number of occurrences */
     int mDuration;
@@ -1789,7 +1788,9 @@
   case rSecondly:
     periods = static_cast<int>( start.secsTo_long( toDate ) / modifier );
     // round it down to the next lower multiple of frequency:
-    periods = ( periods / mFrequency ) * mFrequency;
+    if ( mFrequency > 0 ) {
+      periods = ( periods / mFrequency ) * mFrequency;
+    }
     nextValid = start.addSecs( modifier * periods );
     break;
   case rWeekly:
@@ -1799,7 +1800,9 @@
   case rDaily:
     periods = start.daysTo( toDate ) / modifier;
     // round it down to the next lower multiple of frequency:
-    periods = ( periods / mFrequency ) * mFrequency;
+    if ( mFrequency > 0 ) {
+      periods = ( periods / mFrequency ) * mFrequency;
+    }
     nextValid = start.addDays( modifier * periods );
     break;
   case rMonthly:
@@ -1807,7 +1810,9 @@
     periods = 12 * ( toDate.date().year() - start.date().year() ) +
               ( toDate.date().month() - start.date().month() );
     // round it down to the next lower multiple of frequency:
-    periods = ( periods / mFrequency ) * mFrequency;
+    if ( mFrequency > 0 ) {
+      periods = ( periods / mFrequency ) * mFrequency;
+    }
     // set the day to the first day of the month, so we don't have problems
     // with non-existent days like Feb 30 or April 31
     start.setDate( QDate( start.date().year(), start.date().month(), 1 ) );
@@ -1816,7 +1821,9 @@
   case rYearly:
     periods = ( toDate.date().year() - start.date().year() );
     // round it down to the next lower multiple of frequency:
-    periods = ( periods / mFrequency ) * mFrequency;
+    if ( mFrequency > 0 ) {
+      periods = ( periods / mFrequency ) * mFrequency;
+    }
     nextValid.setDate( start.date().addYears( periods ) );
     break;
   default:
@@ -1853,7 +1860,7 @@
   case rSecondly:
     periods = static_cast<int>( start.secsTo_long( toDate ) / modifier );
     periods = qMax( 0L, periods );
-    if ( periods > 0 ) {
+    if ( periods > 0 && mFrequency > 0 ) {
       periods += ( mFrequency - 1 - ( ( periods - 1 ) % mFrequency ) );
     }
     nextValid = start.addSecs( modifier * periods );
@@ -1866,7 +1873,7 @@
   case rDaily:
     periods = start.daysTo( toDate ) / modifier;
     periods = qMax( 0L, periods );
-    if ( periods > 0 ) {
+    if ( periods > 0 && mFrequency > 0 ) {
       periods += ( mFrequency - 1 - ( ( periods - 1 ) % mFrequency ) );
     }
     nextValid = start.addDays( modifier * periods );
@@ -1876,7 +1883,7 @@
     periods = 12 * ( toDate.date().year() - start.date().year() ) +
               ( toDate.date().month() - start.date().month() );
     periods = qMax( 0L, periods );
-    if ( periods > 0 ) {
+    if ( periods > 0 && mFrequency > 0 ) {
       periods += ( mFrequency - 1 - ( ( periods - 1 ) % mFrequency ) );
     }
     // set the day to the first day of the month, so we don't have problems
@@ -1888,7 +1895,7 @@
   case rYearly:
     periods = ( toDate.date().year() - start.date().year() );
     periods = qMax( 0L, periods );
-    if ( periods > 0 ) {
+    if ( periods > 0 && mFrequency > 0 ) {
       periods += ( mFrequency - 1 - ( ( periods - 1 ) % mFrequency ) );
     }
     nextValid.setDate( start.date().addYears( periods ) );
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ CMakeLists.txt	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -5,11 +5,11 @@
 # Used e.g. in KdepimLibsConfig.cmake, Alex
 set(KDEPIMLIBS_VERSION_MAJOR 4)
 set(KDEPIMLIBS_VERSION_MINOR 2)
-set(KDEPIMLIBS_VERSION_PATCH 3)
+set(KDEPIMLIBS_VERSION_PATCH 4)
 set(KDEPIMLIBS_VERSION ${KDEPIMLIBS_VERSION_MAJOR}.${KDEPIMLIBS_VERSION_MINOR}.${KDEPIMLIBS_VERSION_PATCH} )
 
 # search packages used by KDE
-find_package(KDE4  4.2.3  REQUIRED)
+find_package(KDE4  4.2.4  REQUIRED)
 
 include (KDE4Defaults)
 include (MacroLibrary)
Index: kioslave/smtp/smtp.cpp
===================================================================
--- kioslave/smtp/smtp.cpp	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kioslave/smtp/smtp.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -132,9 +132,6 @@
   s >> what;
   if ( what == 'c' ) {
     infoMessage( createSpecialResponse() );
-#ifndef NDEBUG
-    kDebug(7112) << "special('c') returns \"" << createSpecialResponse() << "\"";
-#endif
   } else if ( what == 'N' ) {
     if ( !execute( Command::NOOP ) )
       return;
Index: kabc/plugins/file/file.desktop
===================================================================
--- kabc/plugins/file/file.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kabc/plugins/file/file.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -17,6 +17,7 @@
 Name[km]=ឯកសារ
 Name[lt]=Failas
 Name[lv]=Fails
+Name[mai]=फाइल
 Name[nb]=Fil
 Name[nds]=Datei
 Name[nl]=Bestand
@@ -41,6 +42,7 @@
 Name[zh_TW]=檔案
 Comment=Provides access to contacts stored in a single local file. Supports standard VCard files and other formats depending on available plugins
 Comment[ca]=Proporciona l'accés als contactes emmagatzemats en un fitxer individual. Accepta els fitxers estàndard VCard i altres formats, depenent dels connectors disponibles
+Comment[cs]=Poskytuje přístup ke kontaktům uloženým v jednom lokálním souboru. Podporuje standardní soubory s vizitkou a jiným formátům podle dostupných modulů
 Comment[da]=Giver adgang til kontakter, hver lagret i en enkelt fil. Understøtter standard VCard-fil og andre formater afhængigt af tilgængelige af plugins
 Comment[de]=Ermöglicht Zugriff auf Kontakte, die in einer einzigen Dateien lokal gespeichert sind. Unterstützt Standard-VCard-Dateien und andere Formate abhängig von den verfügbaren Modulen.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές, αποθηκευμένες σε ένα τοπικό αρχείο. Υποστηρίζει  τυπικά αρχεία VCard και άλλες μορφές αρχείων ανάλογα με τη διαθεσιμότητα των πρόσθετων
Index: kabc/plugins/ldapkio/ldapkio.desktop
===================================================================
--- kabc/plugins/ldapkio/ldapkio.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kabc/plugins/ldapkio/ldapkio.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -6,6 +6,7 @@
 Name[x-test]=xxLDAPxx
 Comment=Provides access to contacts stored on a LDAP directory server
 Comment[ca]=Proporciona l'accés als contactes emmagatzemats en un servidor de directori LDAP
+Comment[cs]=Poskytuje přístup ke kontaktům uloženým v adresářovém serveru LDAP
 Comment[da]=Giver adgang til kontakter lagret på en LDAP directory-server
 Comment[de]=Ermöglicht Zugriff auf Kontakte, die auf einem LDAP-Server gespeichert sind.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές, αποθηκευμένες σε έναν εξυπηρετητή καταλόγου LDAP
Index: kabc/plugins/net/net.desktop
===================================================================
--- kabc/plugins/net/net.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kabc/plugins/net/net.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -18,6 +18,7 @@
 Name[km]=បណ្តាញ
 Name[lt]=Tinklas
 Name[lv]=Tīkls
+Name[mai]=नेटवर्क
 Name[nb]=Nettverk
 Name[nds]=Nettwark
 Name[nl]=Netwerk
@@ -42,6 +43,7 @@
 Name[zh_TW]=網路
 Comment=Provides access to contacts in remote files using KDE's network framework KIO. Supports standard VCard files and other formats depending on available plugins
 Comment[ca]=Proporciona l'accés als contactes en fitxers remots usant la infraestructura KIO de xarxa del KDE. Accepta els fitxers estàndard VCard i altres formats, depenent dels connectors disponibles
+Comment[cs]=Poskytuje přístup ke vzdáleným kontaktům pomocí síťového frameworku KDE KIO. Podporuje standardní soubory s vizitkou a jiným formátům podle dostupných modulů
 Comment[da]=Giver adgang til kontakter i eksterne filer med brug af KDE's netværks-framework KIO. Understøtter standard VCard-filer og andre formater afhængigt af tilgængelige plugins
 Comment[de]=Ermöglicht Zugriff auf Kontakte in entfernten Dateien durch das KIO-Netzwerksystem von KDE. Unterstützt Standard-VCard-Dateien und andere Formate abhängig von den verfügbaren Modulen.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές σε απομακρυσμένα αρχεία με τη χρήση του συστήματος KIO  του KDE. Υποστηρίζει τυπικά αρχεία VCard και άλλες μορφές αρχείων ανάλογα με τη  διαθεσιμότητα των πρόσθετων
Index: kabc/plugins/dir/dir.desktop
===================================================================
--- kabc/plugins/dir/dir.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kabc/plugins/dir/dir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -18,6 +18,7 @@
 Name[km]=ថត
 Name[lt]=Aplankas
 Name[lv]=Mape
+Name[mai]=निर्देशिका
 Name[nb]=Mappe
 Name[nds]=Orner
 Name[nl]=Map
@@ -42,6 +43,7 @@
 Name[zh_TW]=目錄
 Comment=Provides access to contacts, each stored in a single file, in a given directory. Supports standard VCard file and other formats depending on availability of plugins
 Comment[ca]=Proporciona l'accés als contactes, cada un emmagatzemat en un fitxer individual, en un directori donat. Accepta els fitxers estàndard VCard i altres formats depenent de la disponibilitat dels connectors
+Comment[cs]=Poskytuje přístup ke kontaktům, každém uloženém v jednom souboru v daném adresáři. Podporuje standardní soubory s vizitkou a jiným formátům podle dostupných modulů
 Comment[da]=Giver adgang til kontakter, hver lagret i en enkelt fil, i en given mappe. Understøtter standard VCard-fil og andre formater afhængigt af tilgængeligheden af plugins
 Comment[de]=Ermöglicht Zugriff auf Kontakte, die in einzelnen Dateien in einem vorgegebenen Ordner gespeichert sind. Unterstützt Standard-VCard-Dateien und andere Formate abhängig von den verfügbaren Modulen.
 Comment[el]=Προσφέρει πρόσβαση σε επαφές, αποθηκευμένες σε ξεχωριστά αρχεία, σε ένα  δοσμένο κατάλογο. Υποστηρίζει τυπικά αρχεία VCard και άλλες μορφές αρχείων  ανάλογα με τη διαθεσιμότητα των πρόσθετων
Index: kabc/kabc_manager.desktop
===================================================================
--- kabc/kabc_manager.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kabc/kabc_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -18,6 +18,7 @@
 Name[km]=ទំនាក់ទំនង​
 Name[lt]=Kontaktai
 Name[lv]=Kontakti
+Name[mai]=संपर्क
 Name[nb]=Kontakter
 Name[nds]=Kontakten
 Name[nl]=Contactpersonen
Index: kabc/formats/binary.desktop
===================================================================
--- kabc/formats/binary.desktop	(.../tags/KDE/4.2.3/kdepimlibs)	(wersja 974231)
+++ kabc/formats/binary.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 974231)
@@ -18,6 +18,7 @@
 Name[km]=គោលពីរ
 Name[lt]=Dvejetainis
 Name[lv]=Binārs
+Name[mai]=द्विपदीय
 Name[nb]=Binært
 Name[nds]=Bineer
 Name[nl]=Binair
