--- akonadi/CMakeLists.txt	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/CMakeLists.txt-new	2008-05-09 05:35:13.000000000 +0200
@@ -95,14 +95,14 @@
 )
 
 # DBus interfaces and adaptors
-set(akonadi_xml ${AKONADI_DBUS_INTERFACES_DIR}/org.freedesktop.Akonadi.NotificationManager.xml)
+set(akonadi_xml ${AKONADI_DBUS_INTERFACES_DIR}/org.kde.Akonadi.NotificationManager.xml)
 set_source_files_properties(${akonadi_xml} PROPERTIES INCLUDE "notificationmessage_p.h")
 qt4_add_dbus_interface( akonadikde_LIB_SRC ${akonadi_xml} notificationmanagerinterface )
-qt4_add_dbus_interfaces( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.freedesktop.Akonadi.AgentManager.xml )
-qt4_add_dbus_interfaces( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.freedesktop.Akonadi.Tracer.xml )
-qt4_add_dbus_adaptor( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.freedesktop.Akonadi.Resource.xml resourcebase.h Akonadi::ResourceBase )
-qt4_add_dbus_adaptor( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.freedesktop.Akonadi.Agent.Status.xml agentbase.h Akonadi::AgentBase )
-qt4_add_dbus_adaptor( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.freedesktop.Akonadi.Agent.Control.xml agentbase.h Akonadi::AgentBase )
+qt4_add_dbus_interfaces( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.kde.Akonadi.AgentManager.xml )
+qt4_add_dbus_interfaces( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.kde.Akonadi.Tracer.xml )
+qt4_add_dbus_adaptor( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.kde.Akonadi.Resource.xml resourcebase.h Akonadi::ResourceBase )
+qt4_add_dbus_adaptor( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.kde.Akonadi.Agent.Status.xml agentbase.h Akonadi::AgentBase )
+qt4_add_dbus_adaptor( akonadikde_LIB_SRC ${AKONADI_DBUS_INTERFACES_DIR}/org.kde.Akonadi.Agent.Control.xml agentbase.h Akonadi::AgentBase )
 
 kde4_add_ui_files( akonadikde_LIB_SRC
   cachepolicypage.ui
--- akonadi/agentbase_p.h	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/agentbase_p.h-new	2008-05-09 05:40:09.000000000 +0200
@@ -81,7 +81,7 @@
 
     ChangeRecorder *mMonitor;
 
-    org::freedesktop::Akonadi::Tracer *mTracer;
+    org::kde::Akonadi::Tracer *mTracer;
 
     AgentBase::Observer *mObserver;
 
--- akonadi/monitor_p.h	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/monitor_p.h-new	2008-05-09 05:42:23.000000000 +0200
@@ -47,7 +47,7 @@
 
     Monitor *q_ptr;
     Q_DECLARE_PUBLIC( Monitor )
-    org::freedesktop::Akonadi::NotificationManager *nm;
+    org::kde::Akonadi::NotificationManager *nm;
     Collection::List collections;
     QSet<QByteArray> resources;
     QSet<Item::Id> items;
--- akonadi/agentbase.cpp	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/agentbase.cpp-new	2008-05-09 05:45:15.000000000 +0200
@@ -137,7 +137,7 @@
    */
   SessionPrivate::createDefaultSession( mId.toLatin1() );
 
-  mTracer = new org::freedesktop::Akonadi::Tracer( QLatin1String( "org.freedesktop.Akonadi" ), QLatin1String( "/tracing" ),
+  mTracer = new org::kde::Akonadi::Tracer( QLatin1String( "org.kde.Akonadi" ), QLatin1String( "/tracing" ),
                                            QDBusConnection::sessionBus(), q );
 
   new ControlAdaptor( q );
@@ -178,7 +178,7 @@
 
 void AgentBasePrivate::delayedInit()
 {
-  if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.freedesktop.Akonadi.Agent." ) + mId ) )
+  if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.kde.Akonadi.Agent." ) + mId ) )
     kFatal() << "Unable to register service at dbus:" << QDBusConnection::sessionBus().lastError().message();
 }
 
@@ -402,12 +402,12 @@
 
 WId AgentBase::winIdForDialogs() const
 {
-  bool registered = QDBusConnection::sessionBus().interface()->isServiceRegistered( QLatin1String("org.freedesktop.akonaditray") );
+  bool registered = QDBusConnection::sessionBus().interface()->isServiceRegistered( QLatin1String("org.kde.akonaditray") );
   if ( !registered )
     return 0;
 
-  QDBusInterface dbus( QLatin1String("org.freedesktop.akonaditray"), QLatin1String("/Actions"),
-                       QLatin1String("org.freedesktop.Akonadi.Tray") );
+  QDBusInterface dbus( QLatin1String("org.kde.akonaditray"), QLatin1String("/Actions"),
+                       QLatin1String("org.kde.Akonadi.Tray") );
   QDBusMessage reply = dbus.call( QLatin1String("getWinId") );
 
   if ( reply.type() == QDBusMessage::ErrorMessage )
--- akonadi/agentmanager.cpp	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/agentmanager.cpp-new	2008-05-09 05:48:19.000000000 +0200
@@ -206,7 +206,7 @@
 AgentManager::AgentManager()
   : QObject( 0 ), d( new AgentManagerPrivate( this ) )
 {
-  d->mManager = new org::freedesktop::Akonadi::AgentManager( QLatin1String( "org.freedesktop.Akonadi.Control" ),
+  d->mManager = new org::kde::Akonadi::AgentManager( QLatin1String( "org.kde.Akonadi.Control" ),
                                                      QLatin1String( "/AgentManager" ),
                                                      QDBusConnection::sessionBus(), this );
 
--- akonadi/agentmanager_p.h	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/agentmanager_p.h-new	2008-05-09 05:49:42.000000000 +0200
@@ -69,7 +69,7 @@
     static AgentManager *mSelf;
 
     AgentManager *mParent;
-    org::freedesktop::Akonadi::AgentManager *mManager;
+    org::kde::Akonadi::AgentManager *mManager;
 
     QHash<QString, AgentType> mTypes;
     QHash<QString, AgentInstance> mInstances;
--- akonadi/control.cpp	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/control.cpp-new	2008-05-09 05:51:22.000000000 +0200
@@ -28,8 +28,8 @@
 #include <QtDBus/QDBusConnection>
 #include <QtDBus/QDBusConnectionInterface>
 
-#define AKONADI_CONTROL_SERVICE QLatin1String("org.freedesktop.Akonadi.Control")
-#define AKONADI_SERVER_SERVICE QLatin1String("org.freedesktop.Akonadi")
+#define AKONADI_CONTROL_SERVICE QLatin1String("org.kde.Akonadi.Control")
+#define AKONADI_SERVER_SERVICE QLatin1String("org.kde.Akonadi")
 
 using namespace Akonadi;
 
--- akonadi/itemserializerplugin.h	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/itemserializerplugin.h-new	2008-05-09 05:52:30.000000000 +0200
@@ -103,6 +103,6 @@
 
 }
 
-Q_DECLARE_INTERFACE( Akonadi::ItemSerializerPlugin, "org.freedesktop.Akonadi.ItemSerializerPlugin/1.0" )
+Q_DECLARE_INTERFACE( Akonadi::ItemSerializerPlugin, "org.kde.Akonadi.ItemSerializerPlugin/1.0" )
 
 #endif
--- akonadi/monitor_p.cpp	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/monitor_p.cpp-new	2008-05-09 05:53:58.000000000 +0200
@@ -45,7 +45,7 @@
   NotificationMessage::registerDBusTypes();
 
   if ( !nm )
-    nm = new org::freedesktop::Akonadi::NotificationManager( QLatin1String( "org.freedesktop.Akonadi" ),
+    nm = new org::kde::Akonadi::NotificationManager( QLatin1String( "org.kde.Akonadi" ),
                                                      QLatin1String( "/notifications" ),
                                                      QDBusConnection::sessionBus(), q_ptr );
   else
--- akonadi/resourcebase.cpp	2008-05-07 11:05:13.000000000 +0200
+++ akonadi/resourcebase.cpp-new	2008-05-09 05:54:56.000000000 +0200
@@ -66,7 +66,7 @@
 
     void delayedInit()
     {
-      if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.freedesktop.Akonadi.Resource." ) + mId ) )
+      if ( !QDBusConnection::sessionBus().registerService( QLatin1String( "org.kde.Akonadi.Resource." ) + mId ) )
         kFatal() << "Unable to register service at D-Bus: " << QDBusConnection::sessionBus().lastError().message();
       AgentBasePrivate::delayedInit();
     }
--- akonadi/itemfetchjob.cpp	2008-05-09 06:07:01.000000000 +0200
+++ akonadi/itemfetchjob.cpp-new	2008-05-09 06:07:08.000000000 +0200
@@ -211,7 +211,7 @@
           int version = 0;
           QByteArray plainKey( key );
 
-          ImapParser::splitVersionedKey( key, plainKey, version );
+          //ImapParser::splitVersionedKey( key, plainKey, version );
 
           ItemSerializer::deserialize( item, plainKey, fetchResponse.value( i + 1 ), version );
         }
