Index: kcal/incidencebase.cpp
===================================================================
--- kcal/incidencebase.cpp	(.../tags/KDE/4.3.4/kdepimlibs)	(wersja 1061285)
+++ kcal/incidencebase.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1061285)
@@ -477,7 +477,9 @@
     d->mUpdatedPending = true;
   } else {
     foreach ( IncidenceObserver *o, d->mObservers ) {
-      o->incidenceUpdated( this );
+      if ( o ) {
+        o->incidenceUpdated( this );
+      }
     }
   }
 }
Index: akonadi/CMakeLists.txt
===================================================================
--- akonadi/CMakeLists.txt	(.../tags/KDE/4.3.4/kdepimlibs)	(wersja 1061285)
+++ akonadi/CMakeLists.txt	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1061285)
@@ -144,11 +144,11 @@
 target_link_libraries( akonadi-kde ${KDE4_SOLID_LIBS} ${QT_QTNETWORK_LIBRARY} ${QT_QTDBUS_LIBRARY} ${QT_QTSQL_LIBRARY} ${KDE4_KDEUI_LIBS} ${KDE4_KIO_LIBS} ${AKONADI_COMMON_LIBRARIES} )
 set( AKONADI_KDE_DEPS ${KDE4_KDEUI_LIBS} ${QT_QTDBUS_LIBRARY} ${QT_QTCORE_LIBRARY} )
 
-if(${KDE_IS_AT_LEAST_42})
+if(KDE_IS_AT_LEAST_42)
 target_link_libraries( akonadi-kde LINK_INTERFACE_LIBRARIES ${AKONADI_KDE_DEPS})
-else(${KDE_IS_AT_LEAST_42})
+else(KDE_IS_AT_LEAST_42)
 target_link_libraries( akonadi-kde ${AKONADI_KDE_DEPS})
-endif(${KDE_IS_AT_LEAST_42})
+endif(KDE_IS_AT_LEAST_42)
 
 
 set_target_properties( akonadi-kde PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
