--- akonadi/CMakeLists.txt.old	2009-12-02 11:21:52.356239133 +0100
+++ akonadi/CMakeLists.txt	2009-12-02 11:21:59.451551081 +0100
@@ -234,7 +234,7 @@
   entitylistview.h
   entitymimetypefiltermodel.h
 # Made private in 4.4 -- Stephen Kelly.
-#  entityrightsfiltermodel.h
+  entityrightsfiltermodel.h
   entitytreemodel.h
   entitytreeview.h
   entitytreeviewstatesaver.h
--- akonadi/CMakeLists.txt.old	2009-12-02 12:21:23.929607077 +0100
+++ akonadi/CMakeLists.txt	2009-12-02 12:21:44.327633304 +0100
@@ -256,6 +256,7 @@
   itemsync.h
   itemview.h
   job.h
+  krecursivefilterproxymodel.h
   linkjob.h
   mimetypechecker.h
   monitor.h
