Index: kioslave/imap4/imap4.cpp
===================================================================
--- kioslave/imap4/imap4.cpp	(.../tags/KDE/4.6.0/kdepimlibs)	(wersja 1217346)
+++ kioslave/imap4/imap4.cpp	(.../branches/KDE/4.6/kdepimlibs)	(wersja 1217346)
@@ -2576,7 +2576,7 @@
     ssize_t copyLen = (len < readBufferLen) ? len : readBufferLen;
     memcpy(data, readBuffer, copyLen);
     readBufferLen -= copyLen;
-    if (readBufferLen) memcpy(readBuffer, &readBuffer[copyLen], readBufferLen);
+    if (readBufferLen) memmove(readBuffer, &readBuffer[copyLen], readBufferLen);
     return copyLen;
   }
   if (!isConnected()) return 0;
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.6.0/kdepimlibs)	(wersja 1217346)
+++ CMakeLists.txt	(.../branches/KDE/4.6/kdepimlibs)	(wersja 1217346)
@@ -6,8 +6,8 @@
 
 ############### The kdepimlibs version (used e.g. in KdepimLibsConfig.cmake) ###############
 set(KDEPIMLIBS_VERSION_MAJOR 4)
-set(KDEPIMLIBS_VERSION_MINOR 5)
-set(KDEPIMLIBS_VERSION_PATCH 95)
+set(KDEPIMLIBS_VERSION_MINOR 6)
+set(KDEPIMLIBS_VERSION_PATCH 00)
 set(KDEPIMLIBS_VERSION ${KDEPIMLIBS_VERSION_MAJOR}.${KDEPIMLIBS_VERSION_MINOR}.${KDEPIMLIBS_VERSION_PATCH})
 
 
