Index: kpimutils/linklocator.cpp
===================================================================
--- kpimutils/linklocator.cpp	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kpimutils/linklocator.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -107,15 +107,19 @@
 {
   QString url;
   if ( atUrl() ) {
-    // handle cases like this: <link>http://foobar.org/</link>
+    // for reference: rfc1738:
+    // Thus, only alphanumerics, the special characters "$-_.+!*'(),", and
+    // reserved characters used for their reserved purposes may be used
+    // unencoded within a URL.
+    // NOTE: this implementation is not RFC conforming
     int start = mPos;
     while ( mPos < (int)mText.length() &&
             mText[mPos] > ' ' && mText[mPos] != '"' &&
-            QString( "<>()[]" ).indexOf( mText[mPos] ) == -1 ) {
+            QString( "<>[]" ).indexOf( mText[mPos] ) == -1 ) {
       ++mPos;
     }
 
-    /* some URLs really end with:  # / & - _    */
+    // some URLs really end with:  # / & - _
     const QString allowedSpecialChars = QString( "#/&-_" );
     while ( mPos > start && mText[mPos-1].isPunct() &&
             allowedSpecialChars.indexOf( mText[mPos-1] ) == -1 ) {
@@ -268,20 +272,38 @@
     ch = locator.mText[locator.mPos];
     if ( flags & PreserveSpaces ) {
       if ( ch == ' ' ) {
+        if ( locator.mPos + 1 < locator.mText.length() ) {
+          if ( locator.mText[locator.mPos + 1] != ' ' ) {
+
+            // A single space, make it breaking if not at the start or end of the line
+            const bool endOfLine = locator.mText[locator.mPos + 1] == '\n';
+            if ( !startOfLine && !endOfLine )
+              result += ' ';
+            else
+              result += "&nbsp;";
+          }
+          else {
+
+            // Whitespace of more than one space, make it all non-breaking
+            while( locator.mPos < locator.mText.length() && locator.mText[locator.mPos] == ' ' ) {
+              result += "&nbsp;";
+              locator.mPos++;
+              x++;
+            }
+
+            // We incremented once to often, undo that
+            locator.mPos--;
+            x--;
+          }
+        }
+        else {
+          // Last space in the text, it is non-breaking
+          result += "&nbsp;";
+        }
+
         if ( startOfLine ) {
-          result += "&nbsp;";
-          locator.mPos++, x++;
           startOfLine = false;
         }
-        while ( locator.mText[locator.mPos] == ' ' ) {
-          result += ' ';
-          locator.mPos++, x++;
-          if ( locator.mText[locator.mPos] == ' ' ) {
-            result += "&nbsp;";
-            locator.mPos++, x++;
-          }
-        }
-        locator.mPos--, x--;
         continue;
       } else if ( ch == '\t' ) {
         do
Index: kpimutils/tests/testlinklocator.cpp
===================================================================
--- kpimutils/tests/testlinklocator.cpp	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kpimutils/tests/testlinklocator.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -19,6 +19,7 @@
   Boston, MA 02110-1301, USA.
 */
 #include <qtest_kde.h>
+#include <kdebug.h>
 
 #include "testlinklocator.h"
 #include "testlinklocator.moc"
@@ -104,3 +105,126 @@
   QVERIFY( llq.getEmailAddress() == "foo@bar-bar.baz" );
 }
 
+void LinkLocatorTest::testGetUrl()
+{
+  QStringList brackets;
+  brackets << "" << "";   // no brackets
+  brackets << "(" << ")";
+  brackets << "<" << ">";
+  brackets << "[" << "]";
+  brackets << "<link>" << "</link>";
+
+  for (int i = 0; i < brackets.count(); i += 2)
+    testGetUrl2(brackets[i], brackets[i+1]);
+}
+
+void LinkLocatorTest::testGetUrl2(const QString &left, const QString &right)
+{
+  QStringList schemas;
+  schemas << "http://";
+  schemas << "https://";
+  schemas << "vnc://";
+  schemas << "fish://";
+  schemas << "ftp://";
+  schemas << "ftps://";
+  schemas << "sftp://";
+  schemas << "smb://";
+  schemas << "file://";
+
+  QStringList urls;
+  urls << "www.kde.org";
+  urls << "user@www.kde.org";
+  urls << "user:pass@www.kde.org";
+  urls << "user:pass@www.kde.org:1234";
+  urls << "user:pass@www.kde.org:1234/sub/path";
+  urls << "user:pass@www.kde.org:1234/sub/path?a=1";
+  urls << "user:pass@www.kde.org:1234/sub/path?a=1#anchor";
+  urls << "user:pass@www.kde.org:1234/sub/path/special(123)?a=1#anchor";
+  urls << "user:pass@www.kde.org:1234/sub/path:with:colon/special(123)?a=1#anchor";
+
+  foreach (QString schema, schemas)
+  {
+    foreach (QString url, urls)
+    {
+      QString test(left + schema + url + right);
+      LinkLocator ll(test, left.length());
+      QString gotUrl = ll.getUrl();
+
+      bool ok = ( gotUrl == (schema + url) );
+      //qDebug() << "check:" << (ok ? "OK" : "NOK") << test << "=>" << (schema + url);
+      QVERIFY2( ok, qPrintable(test) );
+    }
+  }
+
+  QStringList urlsWithoutSchema;
+  urlsWithoutSchema << ".kde.org";
+  urlsWithoutSchema << ".kde.org:1234/sub/path";
+  urlsWithoutSchema << ".kde.org:1234/sub/path?a=1";
+  urlsWithoutSchema << ".kde.org:1234/sub/path?a=1#anchor";
+  urlsWithoutSchema << ".kde.org:1234/sub/path/special(123)?a=1#anchor";
+  urlsWithoutSchema << ".kde.org:1234/sub/path:with:colon/special(123)?a=1#anchor";
+
+  QStringList starts;
+  starts << "www" << "ftp" << "news:www";
+
+  foreach (QString start, starts)
+  {
+    foreach (QString url, urlsWithoutSchema)
+    {
+      QString test(left + start + url + right);
+      LinkLocator ll(test, left.length());
+      QString gotUrl = ll.getUrl();
+
+      bool ok = ( gotUrl == (start + url) );
+      //qDebug() << "check:" << (ok ? "OK" : "NOK") << test << "=>" << (start + url);
+      QVERIFY2( ok, qPrintable(test) );
+    }
+  }
+
+  // mailto
+  {
+    QString addr = "mailto:test@kde.org";
+    QString test(left + addr + right);
+    LinkLocator ll(test, left.length());
+
+    QString gotUrl = ll.getUrl();
+
+    bool ok = ( gotUrl == addr );
+    //qDebug() << "check:" << (ok ? "OK" : "NOK") << test << "=>" << addr;
+    QVERIFY2( ok, qPrintable(test) );
+  }
+}
+
+void LinkLocatorTest::testHtmlConvert_data()
+{
+  QTest::addColumn<QString>("plainText");
+  QTest::addColumn<int>("flags");
+  QTest::addColumn<QString>("htmlText");
+
+  //QTest::newRow( "" ) << "foo" << 0 << "foo";
+  //QTest::newRow( "" ) << "  foo " << 0 << "  foo ";
+  // Linker error when using PreserveSpaces, therefore the hardcoded 0x01
+  QTest::newRow( "" ) << " foo" << 0x01 << "&nbsp;foo";
+  QTest::newRow( "" ) << "  foo" << 0x01 << "&nbsp;&nbsp;foo";
+  QTest::newRow( "" ) << "  foo  " << 0x01 << "&nbsp;&nbsp;foo&nbsp;&nbsp;";
+  QTest::newRow( "" ) << "  foo " << 0x01 << "&nbsp;&nbsp;foo&nbsp;";
+  QTest::newRow( "" ) << "bla bla bla bla bla" << 0x01 << "bla bla bla bla bla";
+  QTest::newRow( "" ) << "bla bla bla \n  bla bla bla " << 0x01
+                      << "bla bla bla&nbsp;<br />\n&nbsp;&nbsp;bla bla bla&nbsp;";
+  QTest::newRow( "" ) << "bla bla  bla" << 0x01
+                      << "bla bla&nbsp;&nbsp;bla";
+  QTest::newRow( "" ) << " bla bla \n bla bla a\n  bla bla " << 0x01
+                      << "&nbsp;bla bla&nbsp;<br />\n&nbsp;bla bla a<br />\n&nbsp;&nbsp;bla bla&nbsp;";
+}
+
+void LinkLocatorTest::testHtmlConvert()
+{
+  QFETCH(QString, plainText);
+  QFETCH(int, flags);
+  QFETCH(QString, htmlText);
+
+  QString actualHtml = LinkLocator::convertToHtml( plainText, flags );
+  QCOMPARE( actualHtml, htmlText );
+}
+
+
Index: kpimutils/tests/testlinklocator.h
===================================================================
--- kpimutils/tests/testlinklocator.h	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kpimutils/tests/testlinklocator.h	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -29,6 +29,12 @@
   Q_OBJECT
   private Q_SLOTS:
     void testGetEmailAddress();
+    void testGetUrl();
+    void testHtmlConvert();
+    void testHtmlConvert_data();
+
+  private:
+    void testGetUrl2(const QString &left, const QString &right);
 };
 
 #endif
Index: kcal/icalformat_p.cpp
===================================================================
--- kcal/icalformat_p.cpp	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kcal/icalformat_p.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -629,7 +629,7 @@
     if ( !c.key().startsWith( "X-KDE" ) &&          //krazy:exclude=strings
          !c.key().startsWith( "X-LibKCal" ) &&      //krazy:exclude=strings
          !c.key().startsWith( "X-MICROSOFT" ) &&    //krazy:exclude=strings
-         !c.key().startsWith( "X-MOZILLA" ) &&      //krazy:exclude=strings
+         !c.key().startsWith( "X-MOZ" ) &&          //krazy:exclude=strings
          !c.key().startsWith( "X-PILOT" ) ) {       //krazy:exclude=strings
       // use text values for the typical X-FOO property.
       // except for vendor specific X-FOO properties.
Index: kcal/incidenceformatter.cpp
===================================================================
--- kcal/incidenceformatter.cpp	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kcal/incidenceformatter.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -1541,11 +1541,13 @@
   {
     if ( incidence && incidence->revision() > 0 && ( existingIncidence || !helper->calendar() ) ) {
       html += tdOpen;
-      if ( incBase->type() == "Todo" ) {
-        //TODO: 4.4, remove the []
-        html += helper->makeLink( "reply", i18n( "[Enter this into my to-do list]" ) );
-      } else {
-        html += helper->makeLink( "reply", i18n( "[Enter this into my calendar]" ) );
+      if ( incBase ) {
+        if ( incBase->type() == "Todo" ) {
+          //TODO: 4.4, remove the []
+          html += helper->makeLink( "reply", i18n( "[Enter this into my to-do list]" ) );
+        } else {
+          html += helper->makeLink( "reply", i18n( "[Enter this into my calendar]" ) );
+        }
       }
       html += tdClose;
     }
@@ -1589,7 +1591,7 @@
       html += tdClose;
 
       // Check in calendar
-      if ( incBase->type() == "Event" ) {
+      if ( incBase && incBase->type() == "Event" ) {
         html += tdOpen;
         //TODO: 4.4, remove the []
         //TODO: 4.4, change to "Check calendar"
@@ -1613,10 +1615,12 @@
     html += tdOpen;
     //TODO: 4.4, remove the []
     //TODO: 4.4, change string to "Enter this response into my..."
-    if ( incBase->type() == "Todo" ) {
-      html += helper->makeLink( "reply", i18n( "[Enter this into my to-do list]" ) );
-    } else {
-      html += helper->makeLink( "reply", i18n( "[Enter this into my calendar]" ) );
+    if ( incBase ) {
+      if ( incBase->type() == "Todo" ) {
+        html += helper->makeLink( "reply", i18n( "[Enter this into my to-do list]" ) );
+      } else {
+        html += helper->makeLink( "reply", i18n( "[Enter this into my calendar]" ) );
+      }
     }
     html += tdClose;
     break;
@@ -2228,7 +2232,7 @@
                     "Recurs every <numid>%1</numid> months on the %2 %3 until %4",
                     recur->frequency(),
                     dayList[rule.pos() + 31],
-                    calSys->weekDayName( rule.day(),KCalendarSystem::LongDayName ),
+                    calSys->weekDayName( rule.day(), KCalendarSystem::LongDayName ),
                     recurEnd( incidence ) );
       if ( recur->duration() >  0 ) {
         txt += i18nc( "number of occurrences",
@@ -2270,20 +2274,22 @@
   case Recurrence::rYearlyMonth:
   {
     if ( recur->duration() != -1 ) {
-      txt = i18ncp( "Recurs Every N years on month-name [1st|2nd|...]"
-                    " until end-date",
-                    "Recurs yearly on %2 %3 until %4",
-                    "Recurs every %1 years on %2 %3 until %4",
-                    recur->frequency(),
-                    calSys->monthName( recur->yearMonths()[0], recur->startDate().year() ),
-                    dayList[ recur->yearDates()[0] + 31 ],
-                    recurEnd( incidence ) );
-      if ( recur->duration() >  0 ) {
-        txt += i18nc( "number of occurrences",
-                      " (<numid>%1</numid> occurrences)",
-                      recur->duration() );
+      if ( !recur->yearDates().isEmpty() ) {
+        txt = i18ncp( "Recurs Every N years on month-name [1st|2nd|...]"
+                      " until end-date",
+                      "Recurs yearly on %2 %3 until %4",
+                      "Recurs every %1 years on %2 %3 until %4",
+                      recur->frequency(),
+                      calSys->monthName( recur->yearMonths()[0], recur->startDate().year() ),
+                      dayList[ recur->yearDates()[0] + 31 ],
+                      recurEnd( incidence ) );
+        if ( recur->duration() >  0 ) {
+          txt += i18nc( "number of occurrences",
+                        " (<numid>%1</numid> occurrences)",
+                        recur->duration() );
+        }
+        return txt;
       }
-      return txt;
     }
     if ( !recur->yearDates().isEmpty() ) {
       return i18ncp( "Recurs Every N years on month-name [1st|2nd|...]",
Index: kcal/attendee.cpp
===================================================================
--- kcal/attendee.cpp	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kcal/attendee.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -153,6 +153,9 @@
   case InProcess:
     return i18nc( "@item to-do in process of being completed", "In Process" );
     break;
+  case None:
+    return i18nc( "@item event or to-do status unknown", "Unknown" );
+    break;
   }
 }
 
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ CMakeLists.txt	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -42,8 +42,8 @@
 
 if (NOT KDEPIM_ONLY_KLEO)
   #FindAkonadi.cmake is only there for compatibility reasons, but we don't want to use that.
-  find_package(Akonadi 1.1.91 QUIET NO_MODULE)
-  macro_log_feature(Akonadi_FOUND "Akonadi" "Akonadi server (from kdesupport)" "http://pim.kde.org/akonadi" TRUE "1.1.95" "Akonadi is required to build KdepimLibs.")
+  find_package(Akonadi 1.2.0 QUIET NO_MODULE)
+  macro_log_feature(Akonadi_FOUND "Akonadi" "Akonadi server (from kdesupport)" "http://pim.kde.org/akonadi" TRUE "1.2.0" "Akonadi is required to build KdepimLibs.")
 
   find_package(Sasl2)
   macro_log_feature(SASL2_FOUND "cyrus-sasl" "Cyrus SASL API" "http://asg.web.cmu.edu/sasl/sasl-library.html" TRUE "" "Required to support authentication of logins in the IMAP and Sieve kioslaves.")
Index: kioslave/pop3/pop3.cpp
===================================================================
--- kioslave/pop3/pop3.cpp	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kioslave/pop3/pop3.cpp	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -470,7 +470,7 @@
         myReadLine(buf, sizeof(buf) - 1);
 
         // HACK: This assumes fread stops at the first \n and not \r
-        if (strcmp(buf, ".\r\n") == 0) {
+        if ( (buf[0] == 0) || (strcmp(buf, ".\r\n") == 0) ) {
           break;              // End of data
         }
         // sanders, changed -2 to -1 below
@@ -831,7 +831,7 @@
         myReadLine(buf, sizeof(buf) - 1);
 
         // HACK: This assumes fread stops at the first \n and not \r
-        if (strcmp(buf, ".\r\n") == 0) {
+        if ( (buf[0] == 0) || (strcmp(buf, ".\r\n") == 0) ) {
           break;                // End of data
         }
         // sanders, changed -2 to -1 below
Index: kholidays/holidays/holiday_cn
===================================================================
--- kholidays/holidays/holiday_cn	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 0)
+++ kholidays/holidays/holiday_cn	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -0,0 +1,40 @@
+: This file is UTF-8 encoded
+: Holiday file for P.R. China
+: Data from http://en.wikipedia.org/w/index.php?title=Public_holidays_in_the_People%27s_Republic_of_China&oldid=297123230
+: Author: Patrick Nagel <mail@patrick-nagel.net>
+:
+"元旦 (New Year)" weekend on 1 january
+magenta "国际妇女节 (International Women's Day)" on 8 march
+black "植树节 (Arbor Day)" on 12 march
+"劳动节 (Labor Day)" weekend on 1 may
+blue "青年节 (Youth Day)" on 4 may
+cyan "六一儿童节 (Children's Day)" on 1 june
+black "建党节 (CPC Founding Day)" on 1 july
+green "建军节 (Army Day)" on 1 august
+"国庆节 (National Day)" weekend on 1 october length 3 days
+:
+: Lunar calendar based holidays
+: and special free (on weekdays) / work (on weekends) arrangements for 2009
+: yellow for free days, red for working days
+: Data from http://cnreviews.com/life/living-in-china/china_public_holiday_2009_20081210.html
+: and various en.wikipedia.org pages
+:
+yellow "节假日 (day off)" weekend on 2 january 2009
+red "工作日 (working day)" on 4 january 2009
+red "工作日 (working day)" on 24 january 2009
+"春节 (Chinese New Year / Spring Festival)" weekend on 25 january 2009 length 3 days
+yellow "节假日 (day off)" weekend on 28 january 2009 length 3 days
+red "工作日 (working day)" on 1 february 2009
+black "元宵节 (Lantern Festival)" on 9 february 2009
+"清明节 (Qing Ming Festival / Tomb Sweeping Day)" weekend on 4 april 2009
+yellow "节假日 (day off)" weekend on 6 april 2009
+"端午节 (Dragon Boat Festival)" weekend on 28 may 2009
+yellow "节假日 (day off)" weekend on 29 may 2009
+red "工作日 (working day)" on 31 may 2009
+black "七夕节 Lover's Day" on 26 august 2009
+: black "中元节 Ghost Festival" on 3 september 2009
+red "工作日 (working day)" on 27 september 2009
+"中秋节 (Mid-Autumn Festival)" weekend on 3 october 2009
+yellow "节假日 (day off)" weekend on 5 october 2009 length 4 days
+red "工作日 (working day)" on 10 october 2009
+black "重阳节 (Double Ninth Day / Elders' Day)" on 26 october 2009
Index: kholidays/holidays/CMakeLists.txt
===================================================================
--- kholidays/holidays/CMakeLists.txt	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ kholidays/holidays/CMakeLists.txt	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -13,6 +13,7 @@
          holiday_catalan
          holiday_ch
          holiday_cl
+         holiday_cn
          holiday_co
          holiday_cz
          holiday_de
Index: mailtransport/transportjob.h
===================================================================
--- mailtransport/transportjob.h	(.../tags/KDE/4.3.0/kdepimlibs)	(wersja 1013409)
+++ mailtransport/transportjob.h	(.../branches/KDE/4.3/kdepimlibs)	(wersja 1013409)
@@ -46,21 +46,25 @@
 
     /**
       Sets the sender of the mail.
+      @p sender must be the plain email address, not including display name.
     */
     void setSender( const QString &sender );
 
     /**
       Sets the "To" receiver(s) of the mail.
+      @p to must be the plain email address(es), not including display name.
     */
     void setTo( const QStringList &to );
 
     /**
       Sets the "Cc" receiver(s) of the mail.
+      @p cc must be the plain email address(es), not including display name.
     */
     void setCc( const QStringList &cc );
 
     /**
       Sets the "Bcc" receiver(s) of the mail.
+      @p bcc must be the plain email address(es), not including display name.
     */
     void setBcc( const QStringList &bcc );
 
