Index: cmake/modules/FindLibical.cmake
===================================================================
--- cmake/modules/FindLibical.cmake	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ cmake/modules/FindLibical.cmake	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -52,7 +52,7 @@
 
 if(LIBICAL_INCLUDE_DIRS AND LIBICAL_LIBRARIES)
   set(FIND_LIBICAL_VERSION_SOURCE
-    "#include <ical.h>\n int main()\n {\n printf(\"%s\",ICAL_VERSION);return 1;\n }\n")
+    "#include <libical/ical.h>\n int main()\n {\n printf(\"%s\",ICAL_VERSION);return 1;\n }\n")
   set(FIND_LIBICAL_VERSION_SOURCE_FILE ${CMAKE_BINARY_DIR}/CMakeTmp/FindLIBICAL.cxx)
   file(WRITE "${FIND_LIBICAL_VERSION_SOURCE_FILE}" "${FIND_LIBICAL_VERSION_SOURCE}")
 
Index: kpimidentities/identitymanager.cpp
===================================================================
--- kpimidentities/identitymanager.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kpimidentities/identitymanager.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -27,6 +27,7 @@
 
 #include <kemailsettings.h> // for IdentityEntry::fromControlCenter()
 #include <klocale.h>
+#include <kglobal.h>
 #include <kdebug.h>
 #include <kconfig.h>
 #include <kuser.h>
@@ -59,6 +60,7 @@
     : QObject( parent )
 {
   setObjectName( name );
+  KGlobal::locale()->insertCatalog("libkpimidentities");
   new IdentityManagerAdaptor( this );
   QDBusConnection dbus = QDBusConnection::sessionBus();
   const QString dbusPath = newDBusObjectName();
Index: syndication/mapper/itemrdfimpl.cpp
===================================================================
--- syndication/mapper/itemrdfimpl.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ syndication/mapper/itemrdfimpl.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -71,12 +71,12 @@
 QList<PersonPtr> ItemRDFImpl::authors() const
 {
     QList<PersonPtr> list;
-    
+
     QStringList people = m_item.dc().creators();
     people += m_item.dc().contributors();
     QStringList::ConstIterator it = people.constBegin();
     QStringList::ConstIterator end = people.constEnd();
-    
+
     for ( ; it != end; ++it)
     {
         PersonPtr ptr = personFromString(*it);
@@ -99,15 +99,15 @@
     if (!m_item.resource()->isAnon())
         return m_item.resource()->uri();
     else
-        return "hash:" + calcMD5Sum(title() + description() + content());
+        return "hash:" + calcMD5Sum(title() + description() + link() + content());
 }
 
-time_t ItemRDFImpl::datePublished() const 
+time_t ItemRDFImpl::datePublished() const
 {
     return m_item.dc().date();
 }
 
-time_t ItemRDFImpl::dateUpdated() const 
+time_t ItemRDFImpl::dateUpdated() const
 {
     return m_item.dc().date();
 }
Index: syndication/mapper/itemrss2impl.cpp
===================================================================
--- syndication/mapper/itemrss2impl.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ syndication/mapper/itemrss2impl.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -72,14 +72,14 @@
 QList<PersonPtr> ItemRSS2Impl::authors() const
 {
     QList<PersonPtr> list;
-    
+
     PersonPtr ptr = personFromString(m_item.author());
-    
+
     if (!ptr->isNull())
     {
         list.append(ptr);
     }
-    
+
     return list;
 }
 
@@ -93,9 +93,9 @@
     QString guid = m_item.guid();
     if (!guid.isEmpty())
         return guid;
-    
-    return QString("hash:%1").arg(calcMD5Sum(title() 
-            + description() + content()));
+
+    return QString("hash:%1").arg(calcMD5Sum(title()
+            + description() + link() + content()));
 }
 
 time_t ItemRSS2Impl::datePublished() const
@@ -111,33 +111,33 @@
 QList<Syndication::EnclosurePtr> ItemRSS2Impl::enclosures() const
 {
     QList<Syndication::EnclosurePtr> list;
-    
+
     QList<Syndication::RSS2::Enclosure> encs = m_item.enclosures();
-    
+
     for (QList<Syndication::RSS2::Enclosure>::ConstIterator it = encs.constBegin();
          it != encs.constEnd(); ++it)
     {
         EnclosureRSS2ImplPtr impl(new EnclosureRSS2Impl(m_item, *it));
         list.append(impl);
     }
-    
+
     return list;
 }
 
 QList<Syndication::CategoryPtr> ItemRSS2Impl::categories() const
 {
     QList<Syndication::CategoryPtr> list;
-    
+
     QList<Syndication::RSS2::Category> cats = m_item.categories();
     QList<Syndication::RSS2::Category>::ConstIterator it = cats.constBegin();
     QList<Syndication::RSS2::Category>::ConstIterator end = cats.constEnd();
-    
+
     for ( ; it != end; ++it)
     {
         CategoryRSS2ImplPtr impl(new CategoryRSS2Impl(*it));
         list.append(impl);
     }
-    
+
     return list;
 }
 
@@ -175,12 +175,12 @@
 QMultiMap<QString, QDomElement> ItemRSS2Impl::additionalProperties() const
 {
     QMultiMap<QString, QDomElement> ret;
-    
+
     foreach (const QDomElement &i, m_item.unhandledElements())
     {
-        ret.insert(i.namespaceURI() + i.localName(), i); 
+        ret.insert(i.namespaceURI() + i.localName(), i);
     }
-    
+
     return ret;
 }
 
Index: kresources/kresources_manager.desktop
===================================================================
--- kresources/kresources_manager.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kresources/kresources_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -14,6 +14,7 @@
 Name[it]=Gestore di KResources
 Name[ja]=KDE リソースマネージャ
 Name[km]=កម្មវិធី​គ្រប់គ្រង KResources
+Name[lt]=KResursų tvarkytuvė
 Name[lv]=KResources pārvaldnieks
 Name[nb]=KResources-styring
 Name[nds]=KResources-Pleger
@@ -25,6 +26,7 @@
 Name[pt_BR]=Gerenciador do KResource
 Name[ro]=Gestionar KResurse
 Name[ru]=Администратор KResources
+Name[se]=KResources-gieđahalli
 Name[sl]=Upravnik KResources
 Name[sr]=Менаџер к‑ресурса
 Name[sr@latin]=Menadžer k‑resursa
@@ -52,6 +54,7 @@
 Comment[it]=Gestore di KResources
 Comment[ja]=KDE リソースマネージャ
 Comment[km]=កម្មវិធី​គ្រប់គ្រង KResources
+Comment[lt]=KResursų tvarkytuvė
 Comment[lv]=KResources pārvaldnieks
 Comment[nb]=KResources-styring
 Comment[nds]=KDE-Ressourcenpleger
@@ -63,6 +66,7 @@
 Comment[pt_BR]=Gerenciador do KResource
 Comment[ro]=Gestionar KResurse
 Comment[ru]=Администратор KResources
+Comment[se]=KResources-gieđahalli
 Comment[sl]=Upravnik KResources
 Comment[sr]=Менаџер к‑ресурса
 Comment[sr@latin]=Menadžer k‑resursa
Index: kresources/kresources.desktop
===================================================================
--- kresources/kresources.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kresources/kresources.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -36,6 +36,7 @@
 Name[pt_BR]=Fontes de dados do KDE
 Name[ro]=Resurse KDE
 Name[ru]=Ресурсы KDE
+Name[se]=KDE-resurssat
 Name[sl]=Viri KDE
 Name[sr]=КДЕ ресурси
 Name[sr@latin]=KDE resursi
@@ -75,6 +76,7 @@
 Comment[pt_BR]=Configurar as fontes de dados do KDE
 Comment[ro]=Configurare resurse KDE
 Comment[ru]=Настроить ресурсы KDE
+Comment[se]=Heivet KDE-resurssaid
 Comment[sl]=Nastavi vire KDE
 Comment[sr]=Подеси ресурсе КДЕ‑а
 Comment[sr@latin]=Podesi resurse KDE‑a
Index: kresources/kresources_plugin.desktop
===================================================================
--- kresources/kresources_plugin.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kresources/kresources_plugin.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -14,6 +14,7 @@
 Name[it]=Plugin di KResources
 Name[ja]=KDE リソースプラグイン
 Name[km]=កម្មវិធី​ជំនួយ​របស់ KResources
+Name[lt]=KResursų priedas
 Name[lv]=KResources spraudnis
 Name[nb]=Programtillegg for KResources
 Name[nds]=KResources-Moduul
@@ -25,6 +26,7 @@
 Name[pt_BR]=Plug-in do KResource
 Name[ro]=Modul KResurse
 Name[ru]=Модуль KResources
+Name[se]=KResources-lassemodula
 Name[sl]=Vstavek KResources
 Name[sr]=Прикључак радног оквира к‑ресурса
 Name[sr@latin]=Priključak radnog okvira k‑resursa
@@ -52,6 +54,7 @@
 Comment[it]=Plugin di infrastruttura di KResources
 Comment[ja]=KDE リソースフレームワークのプラグイン
 Comment[km]=កម្មវិធី​ជំនួយ​គ្រោងការ​របស់ KResources
+Comment[lt]=KResursų aplinkos priedas
 Comment[lv]=KResource karkasa spraudnis
 Comment[nb]=Programtillegg for KResources-rammeverket
 Comment[nds]=Rahmenwark-Moduul för KResources
Index: kmime/kmime_util.h
===================================================================
--- kmime/kmime_util.h	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kmime/kmime_util.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -57,8 +57,7 @@
 
 inline bool isOfSet( const uchar map[16], unsigned char ch )
 {
-  Q_ASSERT( ch < 128 );
-  return map[ ch/8 ] & 0x80 >> ch%8;
+  return ( ch < 128 ) && ( map[ ch/8 ] & 0x80 >> ch%8 );
 }
 inline bool isSpecial( char ch )
 {
Index: kcal/localdir.desktop
===================================================================
--- kcal/localdir.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/localdir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -24,9 +24,10 @@
 Name[pa]=ਲੋਕਲ ਡਾਇਰੈਕਟਰੀ 'ਚ ਕੈਲੰਡਰ
 Name[pl]=Kalendarz w katalogu lokalnym
 Name[pt]=Calendário numa Pasta Local
-Name[pt_BR]=Calendário em uma pasta local
+Name[pt_BR]=Calendário em pasta local
 Name[ro]=Calendar în director local
 Name[ru]=Календарь в локальной папке
+Name[se]=Kaleandar báikkálaš katalogas
 Name[sl]=Koledar v krajevnem imeniku
 Name[sr]=Календар у локалној фасцикли
 Name[sr@latin]=Kalendar u lokalnoj fascikli
@@ -50,6 +51,7 @@
 Comment[it]=Fornisce accesso a voci di calendario, ciascuna memorizzata in un singolo file in una data cartella
 Comment[ja]=それぞれが単一のファイルに保存されている任意のディレクトリ内のカレンダーアイテムへのアクセスを提供します。
 Comment[km]=ផ្ដល់​ការ​ចូលដំណើរការ​ទៅ​កាន់​ធាតុ​ប្រតិទិន ធាតុ​នីមួយៗ​បាន​ផ្ទុក​នៅ​ក្នុង​ឯកសារ​តែ​មួយ នៅ​ក្នុង​ថត​ដែល​បាន​ផ្ដល់​ឲ្យ
+Comment[lt]=Suteikia prieigą prie kalendoriaus įrašų, kurie saugomi individualiai nurodytame aplanke
 Comment[lv]=Nodrošina piekļuvi kalendāra ierakstiem, kas katrs glabājas vienā failā, norādītā mapē
 Comment[nb]=Gir tilgang til kalenderoppføringer, hver  lagret i en enkelt fil, i en gitt mappe
 Comment[nds]=Stellt Togriep op Kalennerindrääg praat, elkeen binnen een Datei binnen en angeven Orner.
@@ -59,6 +61,7 @@
 Comment[pt]=Oferece o acesso aos itens do calendário, estando cada um guardado num único ficheiro de uma dada pasta
 Comment[pt_BR]=Fornece acesso aos itens do calendário, cada um armazenado em um único arquivo na pasta indicada
 Comment[ro]=Oferă acces la elemente de calendar, fiecare stocat într-un singur fișier, într-un anumit director
+Comment[se]=Dáinna beasat kaleandarmerkošidda, vurkejuvvon okta fiilan, dihto katalogas
 Comment[sl]=Nudi dostop do koledarskih vnosov, ki so shranjeni vsak posebej v svoji datoteki, v danem imeniku
 Comment[sr]=Пружа приступ календарским ставкама, складиштеним у појединачним фајловима у датој фасцикли.
 Comment[sr@latin]=Pruža pristup kalendarskim stavkama, skladištenim u pojedinačnim fajlovima u datoj fascikli.
Index: kcal/local.desktop
===================================================================
--- kcal/local.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/local.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -24,9 +24,10 @@
 Name[pa]=ਲੋਕਲ ਫਾਇਲ 'ਚ ਕੈਲੰਡਰ
 Name[pl]=Kalendarz w pliku lokalnym
 Name[pt]=Calendário num Ficheiro Local
-Name[pt_BR]=Calendário em um arquivo local
+Name[pt_BR]=Calendário em arquivo local
 Name[ro]=Calendar în fișier local
 Name[ru]=Календарь в локальном файле
+Name[se]=Kaleandar báikkálaš fiillas
 Name[sl]=Koledar v krajevni datoteki
 Name[sr]=Календар у локалном фајлу
 Name[sr@latin]=Kalendar u lokalnom fajlu
@@ -50,6 +51,7 @@
 Comment[it]=Fornisce accesso ad un calendario memorizzato in un singolo file locale
 Comment[ja]=単一のローカルファイルに保存されているカレンダーへのアクセスを提供します。
 Comment[km]=ផ្ដល់​ការ​ចូលដំណើរការ​ទៅកាន់​ប្រតិទិន​ដែល​បានផ្ទុក​នៅ​ក្នុង​​ឯកសារ​មូលដ្ឋាន​តែ​មួយ
+Comment[lt]=Suteikia prieigą prie kalendoriaus, esančio viename vietiniame faile
 Comment[lv]=Nodrošina piekļuvi kalendāram, kas glabājas vienā lokālā failā
 Comment[nb]=Gir tilgang til en kalender lagret i en enkelt, lokal fil
 Comment[nds]=Stellt Togriep op en Kalenner binnen een enkel lokaal Datei praat.
@@ -59,6 +61,7 @@
 Comment[pt]=Oferece o acesso a um calendário guardado num único ficheiro local
 Comment[pt_BR]=Fornece acesso a um calendário armazenado em um único arquivo local
 Comment[ro]=Oferă acces la un calendar stocat într-un singur fișier local
+Comment[se]=Dáinna beasat kaleandarii vurkejuvvon okta báikásaš fiilan
 Comment[sl]=Nudi dostop do koledarja, ki je shranjen v eni sami krajevni datoteki
 Comment[sr]=Пружа приступ календару складиштеном у једном локалном фајлу
 Comment[sr@latin]=Pruža pristup kalendaru skladištenom u jednom lokalnom fajlu
Index: kcal/kcal_manager.desktop
===================================================================
--- kcal/kcal_manager.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/kcal_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -28,6 +28,7 @@
 Name[pt]=Calendário
 Name[pt_BR]=Calendário
 Name[ru]=Календарь
+Name[se]=Kaleandar
 Name[sl]=Koledar
 Name[sr]=Календар
 Name[sr@latin]=Kalendar
Index: kcal/calendarresources.cpp
===================================================================
--- kcal/calendarresources.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/calendarresources.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -212,6 +212,7 @@
   : Calendar( timeSpec ),
     d( new KCal::CalendarResources::Private( family ) )
 {
+  mPendingDeleteFromResourceMap = false;
   d->mManager->addObserver( this );
 }
 
@@ -220,6 +221,7 @@
   : Calendar( timeZoneId ),
     d( new KCal::CalendarResources::Private( family ) )
 {
+  mPendingDeleteFromResourceMap = false;
   d->mManager->addObserver( this );
 }
 
@@ -287,6 +289,7 @@
   }
 
   d->mOpen = true;
+  emit calendarLoaded();
 }
 
 bool CalendarResources::reload()
@@ -439,7 +442,7 @@
   if ( d->mResourceMap.find( event ) != d->mResourceMap.end() ) {
     status = d->mResourceMap[event]->deleteEvent( event );
     if ( status ) {
-      d->mResourceMap.remove( event );
+      mPendingDeleteFromResourceMap = true;
     }
   } else {
     status = false;
@@ -448,6 +451,9 @@
       status = (*it)->deleteEvent( event ) || status;
     }
   }
+  if ( status ) {
+    notifyIncidenceDeleted( event );
+  }
 
   setModified( status );
   return status;
@@ -492,7 +498,7 @@
   if ( d->mResourceMap.find( todo ) != d->mResourceMap.end() ) {
     status = d->mResourceMap[todo]->deleteTodo( todo );
     if ( status ) {
-      d->mResourceMap.remove( todo );
+      mPendingDeleteFromResourceMap = true;
     }
   } else {
     CalendarResourceManager::ActiveIterator it;
@@ -638,7 +644,7 @@
   if ( d->mResourceMap.find( journal ) != d->mResourceMap.end() ) {
     status = d->mResourceMap[journal]->deleteJournal( journal );
     if ( status ) {
-      d->mResourceMap.remove( journal );
+      mPendingDeleteFromResourceMap = true;
     }
   } else {
     CalendarResourceManager::ActiveIterator it;
@@ -839,6 +845,7 @@
     }
     d->mResourceMap[ incidence ] = r;
   }
+  mPendingDeleteFromResourceMap = false;
 
   int count = incrementChangeCount( r );
   if ( count == 1 ) {
@@ -864,6 +871,11 @@
 
   int count = decrementChangeCount( r );
 
+  if ( mPendingDeleteFromResourceMap ) {
+    d->mResourceMap.remove( incidence );
+    mPendingDeleteFromResourceMap = false;
+  }
+
   if ( count == 0 ) {
     bool ok = save( d->mTickets[ r ], incidence );
     if ( ok ) {
Index: kcal/calendarresources.h
===================================================================
--- kcal/calendarresources.h	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/calendarresources.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -721,6 +721,7 @@
     Q_DISABLE_COPY( CalendarResources )
     class Private;
     Private *d;
+    bool mPendingDeleteFromResourceMap;
     //@endcond
 };
 
Index: kcal/event.cpp
===================================================================
--- kcal/event.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/event.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -121,8 +121,15 @@
   if ( hasEndDate() ) {
     return d->mDtEnd;
   }
+
   if ( hasDuration() ) {
-    return duration().end( dtStart() );
+    if ( allDay() ) {
+      // For all day events, dtEnd is always inclusive
+      KDateTime end = duration().end( dtStart() ).addDays( -1 );
+      return end >= dtStart() ? end : dtStart();
+    } else {
+      return duration().end( dtStart() );
+    }
   }
 
   kDebug() << "Warning! Event '" << summary()
@@ -149,10 +156,10 @@
       timeZone = ' ' + spec.timeZone().name();
     }
 
-    return KGlobal::locale()->formatTime( dtEnd().toTimeSpec( spec ).time(), shortfmt )
+    return KGlobal::locale()->formatTime( dtEnd().toTimeSpec( spec ).time(), !shortfmt )
       + timeZone;
   } else {
-    return KGlobal::locale()->formatTime( dtEnd().time(), shortfmt );
+    return KGlobal::locale()->formatTime( dtEnd().time(), !shortfmt );
   }
 }
 
Index: kcal/todo.cpp
===================================================================
--- kcal/todo.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/todo.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -184,10 +184,10 @@
       timeZone = ' ' + spec.timeZone().name();
     }
 
-    return KGlobal::locale()->formatTime( dtDue( !recurs() ).toTimeSpec( spec ).time(), shortfmt )
+    return KGlobal::locale()->formatTime( dtDue( !recurs() ).toTimeSpec( spec ).time(), !shortfmt )
       + timeZone;
   } else {
-    return KGlobal::locale()->formatTime( dtDue( !recurs() ).time(), shortfmt );
+    return KGlobal::locale()->formatTime( dtDue( !recurs() ).time(), !shortfmt );
   }
 }
 
@@ -308,10 +308,10 @@
       timeZone = ' ' + spec.timeZone().name();
     }
 
-    return KGlobal::locale()->formatTime( dtStart( first ).toTimeSpec( spec ).time(), shortfmt )
+    return KGlobal::locale()->formatTime( dtStart( first ).toTimeSpec( spec ).time(), !shortfmt )
       + timeZone;
   } else {
-    return KGlobal::locale()->formatTime( dtStart( first ).time(), shortfmt );
+    return KGlobal::locale()->formatTime( dtStart( first ).time(), !shortfmt );
   }
 }
 
Index: kcal/incidencebase.cpp
===================================================================
--- kcal/incidencebase.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/incidencebase.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -252,10 +252,10 @@
       timeZone = ' ' + spec.timeZone().name();
     }
 
-    return KGlobal::locale()->formatTime( dtStart().toTimeSpec( spec ).time(), shortfmt )
+    return KGlobal::locale()->formatTime( dtStart().toTimeSpec( spec ).time(), !shortfmt )
       + timeZone;
   } else {
-    return KGlobal::locale()->formatTime( dtStart().time(), shortfmt );
+    return KGlobal::locale()->formatTime( dtStart().time(), !shortfmt );
   }
 }
 
Index: kcal/icalformat.cpp
===================================================================
--- kcal/icalformat.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/icalformat.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -131,7 +131,7 @@
 
   // Convert to UTF8 and save
   QByteArray textUtf8 = text.toUtf8();
-  file.write( textUtf8.data(), textUtf8.size() - 1 );
+  file.write( textUtf8.data(), textUtf8.size() );
 
   if ( !file.finalize() ) {
     kDebug() << "err:" << file.errorString();
Index: kcal/resourcecalendar.cpp
===================================================================
--- kcal/resourcecalendar.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kcal/resourcecalendar.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -252,7 +252,8 @@
 
 bool ResourceCalendar::doSave( bool syncCache, Incidence *incidence )
 {
-  return doSave( syncCache, incidence );
+  Q_UNUSED( incidence );
+  return doSave( syncCache );
 }
 
 void ResourceCalendar::saveError( const QString &err )
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ CMakeLists.txt	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -9,8 +9,7 @@
 set(KDEPIMLIBS_VERSION ${KDEPIMLIBS_VERSION_MAJOR}.${KDEPIMLIBS_VERSION_MINOR}.${KDEPIMLIBS_VERSION_PATCH} )
 
 # search packages used by KDE
-# Adjust the version of kdelibs which is required as needed (should be 4.2.0 once KDE 4.2 is released), Alex
-find_package(KDE4  4.1.80  REQUIRED)
+find_package(KDE4  4.2.0  REQUIRED)
 
 include (KDE4Defaults)
 include (MacroLibrary)
Index: kioslave/nntp/nntp.cpp
===================================================================
--- kioslave/nntp/nntp.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kioslave/nntp/nntp.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -154,6 +154,7 @@
   while ( true ) {
     if ( !waitForResponse( readTimeout() ) ) {
       error( ERR_SERVER_TIMEOUT, mHost );
+      nntp_close();
       return;
     }
     memset( tmp, 0, MAX_PACKET_LEN );
@@ -380,6 +381,7 @@
   while ( true ) {
     if ( ! waitForResponse( readTimeout() ) ) {
       error( ERR_SERVER_TIMEOUT, mHost );
+      nntp_close();
       return;
     }
     memset( readBuffer, 0, MAX_PACKET_LEN );
@@ -451,6 +453,7 @@
     while ( true ) {
       if ( ! waitForResponse( readTimeout() ) ) {
         error( ERR_SERVER_TIMEOUT, mHost );
+        nntp_close();
         return;
       }
       memset( readBuffer, 0, MAX_PACKET_LEN );
@@ -603,6 +606,7 @@
     while ( true ) {
       if ( ! waitForResponse( readTimeout() ) ) {
         error( ERR_SERVER_TIMEOUT, mHost );
+        nntp_close();
         return false;
       }
       memset( readBuffer, 0, MAX_PACKET_LEN );
@@ -636,6 +640,7 @@
   while ( true ) {
     if ( ! waitForResponse( readTimeout() ) ) {
       error( ERR_SERVER_TIMEOUT, mHost );
+      nntp_close();
       return false;
     }
     memset( readBuffer, 0, MAX_PACKET_LEN );
@@ -716,7 +721,6 @@
   if ( isConnected() ) {
     write( "QUIT\r\n", 6 );
     disconnectFromHost();
-    opened = false;
     isAuthenticated = false;
   }
   mCurrentGroup.clear();
@@ -751,8 +755,6 @@
     }
 
     DBG << "  nntp_open -- greating was read res_code :" << res_code;
-    // let local class know that we are connected
-    opened = true;
 
     res_code = sendCommand("MODE READER");
 
@@ -790,7 +792,7 @@
 {
   int res_code = 0;
 
-  if ( !opened ) {
+  if ( !nntp_open() ) {
     ERR << "NOT CONNECTED, cannot send cmd" << cmd;
     return 0;
   }
@@ -897,6 +899,7 @@
 {
   if ( !waitForResponse( responseTimeout() ) ) {
     error( ERR_SERVER_TIMEOUT , mHost );
+    nntp_close();
     return -1;
   }
   memset( data, 0, MAX_PACKET_LEN );
Index: kioslave/nntp/nntp.h
===================================================================
--- kioslave/nntp/nntp.h	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kioslave/nntp/nntp.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -78,7 +78,7 @@
  private:
    QString mHost, mUser, mPass;
    quint16 m_port, m_defaultPort;
-   bool postingAllowed, opened, isAuthenticated;
+   bool postingAllowed, isAuthenticated;
    char readBuffer[MAX_PACKET_LEN];
    ssize_t readBufferLen;
    /// Current selected newsgroup
Index: kioslave/sieve/sieve.protocol
===================================================================
--- kioslave/sieve/sieve.protocol	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kioslave/sieve/sieve.protocol	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -35,6 +35,7 @@
 Description[pt]=Um 'ioslave' para o protocolo de filtragem de correio Sieve
 Description[pt_BR]=Um ioslave para o protocolo de filtragem de correio Sieve
 Description[ro]=Un „ioslave” pentru protocolul de filtrare a poștei Sieve
+Description[se]=SO-šláva Sieve e-boastasillenprotokolla várás
 Description[sl]=Ioslave za protokol poštnega filtriranja Sieve
 Description[sr]=У/И‑захват за сито, протокол филтрирања поште
 Description[sr@latin]=U/I‑zahvat za sito, protokol filtriranja pošte
Index: mailtransport/smtpsettings.ui
===================================================================
--- mailtransport/smtpsettings.ui	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ mailtransport/smtpsettings.ui	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -1,8 +1,9 @@
-<ui version="4.0" >
- <author>Volker Krause &lt;vkrause@kde.org>, KovoKs &lt;tomalbers@kde.nl></author>
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <author>Volker Krause &lt;vkrause@kde.org&gt;, KovoKs &lt;tomalbers@kde.nl&gt;</author>
  <class>SMTPSettings</class>
- <widget class="QWidget" name="SMTPSettings" >
-  <property name="geometry" >
+ <widget class="QWidget" name="SMTPSettings">
+  <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
@@ -147,21 +148,28 @@
              </layout>
             </item>
             <item>
-             <widget class="KPushButton" name="checkCapabilities" >
-              <property name="text" >
+             <widget class="KPushButton" name="checkCapabilities">
+              <property name="text">
                <string>Check &amp;What the Server Supports</string>
               </property>
              </widget>
             </item>
+            <item>
+             <widget class="QProgressBar" name="checkCapabilitiesProgress">
+              <property name="visible">
+               <bool>false</bool>
+              </property>
+             </widget>
+            </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer>
-           <property name="orientation" >
+           <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
-           <property name="sizeHint" stdset="0" >
+           <property name="sizeHint" stdset="0">
             <size>
              <width>166</width>
              <height>81</height>
@@ -349,36 +357,36 @@
         </layout>
        </item>
        <item>
-        <widget class="KSeparator" name="kseparator_3" />
+        <widget class="KSeparator" name="kseparator_3"/>
        </item>
        <item>
-        <widget class="QCheckBox" name="kcfg_specifyHostname" >
-         <property name="text" >
+        <widget class="QCheckBox" name="kcfg_specifyHostname">
+         <property name="text">
           <string>Sen&amp;d custom hostname to server</string>
          </property>
         </widget>
        </item>
        <item>
-        <layout class="QHBoxLayout" >
+        <layout class="QHBoxLayout">
          <item>
-          <widget class="QLabel" name="hostnameLabel" >
-           <property name="enabled" >
+          <widget class="QLabel" name="hostnameLabel">
+           <property name="enabled">
             <bool>false</bool>
            </property>
-           <property name="text" >
+           <property name="text">
             <string>Hos&amp;tname:</string>
            </property>
-           <property name="alignment" >
+           <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
-           <property name="buddy" >
+           <property name="buddy">
             <cstring>kcfg_localHostname</cstring>
            </property>
           </widget>
          </item>
          <item>
-          <widget class="KLineEdit" name="kcfg_localHostname" >
-           <property name="enabled" >
+          <widget class="KLineEdit" name="kcfg_localHostname">
+           <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
@@ -386,22 +394,22 @@
         </layout>
        </item>
        <item>
-        <widget class="KSeparator" name="kseparator_2" />
+        <widget class="KSeparator" name="kseparator_2"/>
        </item>
        <item>
-        <layout class="QHBoxLayout" >
+        <layout class="QHBoxLayout">
          <item>
-          <widget class="QLabel" name="precommandLabel" >
-           <property name="text" >
+          <widget class="QLabel" name="precommandLabel">
+           <property name="text">
             <string>Precommand:</string>
            </property>
-           <property name="alignment" >
+           <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item>
-          <widget class="KLineEdit" name="kcfg_precommand" />
+          <widget class="KLineEdit" name="kcfg_precommand"/>
          </item>
         </layout>
        </item>
@@ -411,7 +419,7 @@
    </item>
   </layout>
  </widget>
- <layoutfunction spacing="KDialog::spacingHint" margin="KDialog::marginHint" />
+ <layoutfunction spacing="KDialog::spacingHint" margin="KDialog::marginHint"/>
  <customwidgets>
   <customwidget>
    <class>KButtonGroup</class>
@@ -454,11 +462,11 @@
    <receiver>usernameLabel</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>29</x>
      <y>88</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>112</x>
      <y>117</y>
     </hint>
@@ -470,11 +478,11 @@
    <receiver>kcfg_userName</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>35</x>
      <y>85</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>186</x>
      <y>115</y>
     </hint>
@@ -486,11 +494,11 @@
    <receiver>passwordLabel</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>49</x>
      <y>88</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>61</x>
      <y>142</y>
     </hint>
@@ -502,11 +510,11 @@
    <receiver>password</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>43</x>
      <y>89</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>140</x>
      <y>142</y>
     </hint>
@@ -518,11 +526,11 @@
    <receiver>kcfg_storePassword</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>75</x>
      <y>86</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>62</x>
      <y>176</y>
     </hint>
@@ -534,11 +542,11 @@
    <receiver>kcfg_authenticationType</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>82</x>
      <y>89</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>78</x>
      <y>202</y>
     </hint>
@@ -550,11 +558,11 @@
    <receiver>hostnameLabel</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>90</x>
      <y>334</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>81</x>
      <y>357</y>
     </hint>
@@ -566,11 +574,11 @@
    <receiver>kcfg_localHostname</receiver>
    <slot>setEnabled(bool)</slot>
    <hints>
-    <hint type="sourcelabel" >
+    <hint type="sourcelabel">
      <x>185</x>
      <y>327</y>
     </hint>
-    <hint type="destinationlabel" >
+    <hint type="destinationlabel">
      <x>179</x>
      <y>360</y>
     </hint>
Index: mailtransport/configmodule.cpp
===================================================================
--- mailtransport/configmodule.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ mailtransport/configmodule.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -27,7 +27,7 @@
 using namespace MailTransport;
 
 K_PLUGIN_FACTORY( MailTransportConfigFactory, registerPlugin<ConfigModule>(); )
-K_EXPORT_PLUGIN( MailTransportConfigFactory( "mailtransport" ) )
+K_EXPORT_PLUGIN( MailTransportConfigFactory( "mailtransport", "libmailtransport" ) )
 
 ConfigModule::ConfigModule( QWidget * parent, const QVariantList & args ) :
     KCModule( MailTransportConfigFactory::componentData(), parent, args )
Index: mailtransport/kcm_mailtransport.desktop
===================================================================
--- mailtransport/kcm_mailtransport.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ mailtransport/kcm_mailtransport.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -27,6 +27,7 @@
 Name[it]=Trasporto della posta
 Name[ja]=メール送信手段
 Name[km]=ការ​ដឹកជញ្ជូន​សំបុត្រ
+Name[lt]=Pašto programa
 Name[lv]=Pasta transports
 Name[nb]=E-post-transport
 Name[nds]=Överdreegmetood
@@ -38,6 +39,7 @@
 Name[pt_BR]=Transporte de correio
 Name[ro]=Transport poștal
 Name[ru]=Метод отправки
+Name[se]=E-boastta fievrridanvuohki
 Name[sl]=Prenos pošte
 Name[sr]=Транспорт поште
 Name[sr@latin]=Transport pošte
Index: mailtransport/transportmanagementwidget.cpp
===================================================================
--- mailtransport/transportmanagementwidget.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ mailtransport/transportmanagementwidget.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -37,6 +37,7 @@
 TransportManagementWidget::TransportManagementWidget( QWidget *parent )
   : QWidget( parent ), d( new Private )
 {
+  KGlobal::locale()->insertCatalog(QString::fromLatin1("libmailtransport"));
   d->ui.setupUi( this );
 
   d->ui.transportList->setHeaderLabels( QStringList()
Index: kabc/scripts/addressee.src.cpp
===================================================================
--- kabc/scripts/addressee.src.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/scripts/addressee.src.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -398,6 +398,7 @@
     d->mEmails.removeAll( email );
     d->mEmails.prepend( email );
   } else {
+    d->mEmpty = false;
     if ( preferred ) {
       d->mEmails.prepend( email );
     } else {
Index: kabc/plugins/file/file.desktop
===================================================================
--- kabc/plugins/file/file.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/plugins/file/file.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -27,6 +27,7 @@
 Name[pt_BR]=Arquivo
 Name[ro]=Fișier
 Name[ru]=Файл
+Name[se]=Fiila
 Name[sl]=Datoteka
 Name[sr]=Фајл
 Name[sr@latin]=Fajl
Index: kabc/plugins/net/net.desktop
===================================================================
--- kabc/plugins/net/net.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/plugins/net/net.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -28,6 +28,7 @@
 Name[pt_BR]=Rede
 Name[ro]=Rețea
 Name[ru]=Сеть
+Name[se]=Fierbmi
 Name[sl]=Omrežje
 Name[sr]=Мрежа
 Name[sr@latin]=Mreža
Index: kabc/plugins/dir/dir.desktop
===================================================================
--- kabc/plugins/dir/dir.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/plugins/dir/dir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -28,6 +28,7 @@
 Name[pt_BR]=Pasta
 Name[ro]=Director
 Name[ru]=Папка
+Name[se]=Kataloga
 Name[sl]=Imenik
 Name[sr]=Фасцикла
 Name[sr@latin]=Fascikla
Index: kabc/stdaddressbook.cpp
===================================================================
--- kabc/stdaddressbook.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/stdaddressbook.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -166,8 +166,8 @@
   Resource *res = mParent->standardResource();
   if ( !res ) {
     res = manager->createResource( "file" );
-    res->setResourceName( i18n( "Default Addressbook" ) );
     if ( res ) {
+      res->setResourceName( i18n( "Default Addressbook" ) );
       mParent->addResource( res );
     } else {
       kDebug() << "No resource available!!!";
Index: kabc/kabc_manager.desktop
===================================================================
--- kabc/kabc_manager.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/kabc_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -28,6 +28,7 @@
 Name[pt_BR]=Contatos
 Name[ro]=Contacte
 Name[ru]=Контакт
+Name[se]=Oktavuođat
 Name[sl]=Stiki
 Name[sr]=Контакти
 Name[sr@latin]=Kontakti
Index: kabc/formats/binary.desktop
===================================================================
--- kabc/formats/binary.desktop	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ kabc/formats/binary.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -28,6 +28,7 @@
 Name[pt_BR]=Binário
 Name[ro]=Binar
 Name[ru]=Двоичный
+Name[se]=Binára
 Name[sl]=Dvojiško
 Name[sr]=Бинарни
 Name[sr@latin]=Binarni
Index: akonadi/itemdeletejob.cpp
===================================================================
--- akonadi/itemdeletejob.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ akonadi/itemdeletejob.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -20,7 +20,7 @@
 #include "itemdeletejob.h"
 #include "itemmodifyjob.h"
 #include "job_p.h"
-#include "transactionjobs.h"
+#include "transactionsequence.h"
 #include "expungejob.h"
 
 using namespace Akonadi;
@@ -73,20 +73,16 @@
 {
   Q_D( ItemDeleteJob );
 
-  TransactionBeginJob *begin = new TransactionBeginJob( this );
-  addSubjob( begin );
+  TransactionSequence  *transaction = new TransactionSequence( this );
+  connect( transaction, SIGNAL(result(KJob*)), SLOT(jobDone(KJob*)) );
+  addSubjob( transaction );
 
   d->mItem.setFlag( "\\Deleted" );
-  ItemModifyJob* store = new ItemModifyJob( d->mItem, this );
+  ItemModifyJob* store = new ItemModifyJob( d->mItem, transaction );
   store->disableRevisionCheck();
-  addSubjob( store );
 
-  ExpungeJob *expunge = new ExpungeJob( this );
-  addSubjob( expunge );
-
-  TransactionCommitJob *commit = new TransactionCommitJob( this );
-  connect( commit, SIGNAL(result(KJob*)), SLOT(jobDone(KJob*)) );
-  addSubjob( commit );
+  new ExpungeJob( transaction );
+  transaction->commit();
 }
 
 #include "itemdeletejob.moc"
Index: akonadi/resourcebase.cpp
===================================================================
--- akonadi/resourcebase.cpp	(.../tags/KDE/4.2.0/kdepimlibs)	(wersja 930690)
+++ akonadi/resourcebase.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 930690)
@@ -398,7 +398,17 @@
 
 void ResourceBase::cancelTask()
 {
-  d_func()->changeProcessed();
+  Q_D( ResourceBase );
+  switch ( d->scheduler->currentTask().type ) {
+    case ResourceScheduler::FetchItem:
+      itemRetrieved( Item() ); // sends the error reply and
+      break;
+    case ResourceScheduler::ChangeReplay:
+      d->changeProcessed();
+      break;
+    default:
+      d->scheduler->taskDone();
+  }
 }
 
 void ResourceBase::cancelTask( const QString &msg )
