--- kdepimlibs/kioslave/nntp/nntp.cpp	2009/12/14 22:08:32	1062489
+++ kdepimlibs/kioslave/nntp/nntp.cpp	2009/12/14 22:17:59	1062490
@@ -517,6 +517,11 @@
   if (firstSerNum == 0)
     return true;
   first = qMax( first, firstSerNum );
+  if ( lastSerNum < first ) { // No need to fetch anything
+    // note: this also ensure that "lastSerNum - first" is not negative
+    // in the next test (in "unsigned long" computation this leads to an overflow
+    return true;
+  }
   if ( max > 0 && lastSerNum - first > max )
     first = lastSerNum - max + 1;
 
--- kdepimlibs/kioslave/nntp/nntp.cpp	2009/12/14 22:17:59	1062490
+++ kdepimlibs/kioslave/nntp/nntp.cpp	2009/12/14 22:18:02	1062491
@@ -628,8 +628,10 @@
     return true; // no articles selected
   if ( res == 500 )
     notSupported = true; // unknwon command
-  if ( res != 224 )
+  if ( res != 224 ) {
+    unexpected_response( res, "XOVER" );
     return false;
+  }
 
   long msgSize;
   QString name;
