diff -ur kabc.org/address.cpp kabc/address.cpp
--- kabc.org/address.cpp	2008-01-29 10:15:02.000000000 +0100
+++ kabc/address.cpp	2008-03-16 17:32:29.536589046 +0100
@@ -619,7 +619,7 @@
   }
 
   QString mapfile = KGlobal::dirs()->findResource( "data",
-          QLatin1String( "kabc/countrytransl.map" ) );
+          QLatin1String( "kabc/kde4-countrytransl.map" ) );
 
   QFile file( mapfile );
   if ( file.open( QIODevice::ReadOnly ) ) {
@@ -650,7 +650,7 @@
   }
 
   QString mapfile = KGlobal::dirs()->findResource( "data",
-          QLatin1String( "kabc/countrytransl.map" ) );
+          QLatin1String( "kabc/kde4-countrytransl.map" ) );
 
   QFile file( mapfile );
   if ( file.open( QIODevice::ReadOnly ) ) {
diff -ur kabc.org/CMakeLists.txt kabc/CMakeLists.txt
--- kabc.org/CMakeLists.txt	2008-02-21 10:27:41.000000000 +0100
+++ kabc/CMakeLists.txt	2008-03-16 17:43:47.065645467 +0100
@@ -92,7 +92,7 @@
 
 ########### install files ###############
 
-install( FILES countrytransl.map  DESTINATION  ${DATA_INSTALL_DIR}/kabc )
+install( FILES countrytransl.map  DESTINATION  ${DATA_INSTALL_DIR}/kabc RENAME kde4-countrytransl.map )
 install( FILES kabc_manager.desktop  DESTINATION  ${SERVICES_INSTALL_DIR}/kresources )
 install( FILES
    kabc_export.h
Tylko w kabc: CMakeLists.txt~
diff -ur kabc.org/formats/CMakeLists.txt kabc/formats/CMakeLists.txt
--- kabc.org/formats/CMakeLists.txt	2008-01-05 00:53:23.000000000 +0100
+++ kabc/formats/CMakeLists.txt	2008-03-16 17:45:01.065849765 +0100
@@ -16,7 +16,7 @@
 
 ########### install files ###############
 
-install( FILES binary.desktop  DESTINATION  ${DATA_INSTALL_DIR}/kabc/formats )
+install( FILES binary.desktop  DESTINATION  ${DATA_INSTALL_DIR}/kabc/formats RENAME kde4-binary.desktop )
 
 
 
