Index: cmake/modules/FindLibical.cmake
===================================================================
--- cmake/modules/FindLibical.cmake	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ cmake/modules/FindLibical.cmake	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -16,7 +16,6 @@
 
 if (WIN32)
   file(TO_CMAKE_PATH "$ENV{PROGRAMFILES}" _program_FILES_DIR)
-  string(REPLACE "\\" "/" _program_FILES_DIR "${_program_FILES_DIR}")
 endif(WIN32)
 
 set(LIBICAL_FIND_REQUIRED ${Libical_FIND_REQUIRED})
@@ -28,25 +27,21 @@
 endif(LIBICAL_INCLUDE_DIRS AND LIBICAL_LIBRARIES)
 
 #set the root from the LIBICAL_BASE environment
-string(REPLACE "\\" "/" libical_root "$ENV{LIBICAL_BASE}")
+file(TO_CMAKE_PATH "$ENV{LIBICAL_BASE}" libical_root )
 #override the root from LIBICAL_BASE defined to cmake
 if(DEFINED LIBICAL_BASE)
-  string(REPLACE "\\" "/" libical_root ${LIBICAL_BASE})
+  file(TO_CMAKE_PATH "${LIBICAL_BASE}" libical_root )
 endif(DEFINED LIBICAL_BASE)
 
-find_path(LIBICAL_INCLUDE_DIRS NAMES ical.h
-  PATH_SUFFIXES libical
-  PATHS ${libical_root}/include ${_program_FILES_DIR}/libical/include /usr/local/include /usr/include ${KDE4_INCLUDE_DIR}
-  NO_CMAKE_SYSTEM_PATH
+find_path(LIBICAL_INCLUDE_DIRS NAMES libical/ical.h
+  HINTS ${libical_root}/include ${_program_FILES_DIR}/libical/include ${KDE4_INCLUDE_DIR}
 )
 
 find_library(LIBICAL_LIBRARY NAMES ical libical
-  PATHS ${libical_root}/lib ${_program_FILES_DIR}/libical/lib /usr/local/lib /usr/lib ${KDE4_LIB_DIR}
-  NO_CMAKE_SYSTEM_PATH
+  HINTS ${libical_root}/lib ${_program_FILES_DIR}/libical/lib ${KDE4_LIB_DIR}
 )
 find_library(LIBICALSS_LIBRARY NAMES icalss libicalss
-  PATHS ${libical_root}/lib ${_program_FILES_DIR}/libical/lib /usr/local/lib /usr/lib ${KDE4_LIB_DIR}
-  NO_CMAKE_SYSTEM_PATH
+  HINTS ${libical_root}/lib ${_program_FILES_DIR}/libical/lib ${KDE4_LIB_DIR}
 )
 set(LIBICAL_LIBRARIES ${LIBICAL_LIBRARY} ${LIBICALSS_LIBRARY})
 
Index: kresources/kresources_manager.desktop
===================================================================
--- kresources/kresources_manager.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kresources/kresources_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -10,7 +10,9 @@
 Name[fr]=Gestionnaire de ressources
 Name[ga]=Bainisteoir KResources
 Name[gl]=Xestor de KResources
+Name[hne]=केरिसोर्स प्रबंधक
 Name[hu]=KResources-kezelő
+Name[is]=KResources auðlindastjóri
 Name[it]=Gestore di KResources
 Name[ja]=KDE リソースマネージャ
 Name[km]=កម្មវិធី​គ្រប់គ្រង KResources
@@ -50,7 +52,9 @@
 Comment[fr]=Gestionnaire de ressources
 Comment[ga]=Bainisteoir KResources
 Comment[gl]=Xestor de KResources
+Comment[hne]=केरिसोर्स प्रबंधक
 Comment[hu]=KResources-kezelő
+Comment[is]=KResources auðlindastjóri
 Comment[it]=Gestore di KResources
 Comment[ja]=KDE リソースマネージャ
 Comment[km]=កម្មវិធី​គ្រប់គ្រង KResources
Index: kresources/kresources.desktop
===================================================================
--- kresources/kresources.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kresources/kresources.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -20,6 +20,7 @@
 Name[fr]=Ressources KDE
 Name[ga]=Acmhainní KDE
 Name[gl]=Recursos de KDE
+Name[hne]=केडीई संसाधन
 Name[hu]=KDE-erőforrások
 Name[it]=Risorse di KDE
 Name[ja]=KDE リソース
@@ -60,6 +61,7 @@
 Comment[fr]=Configurer les ressources KDE
 Comment[ga]=Cumraigh Acmhainní KDE
 Comment[gl]=Configurar os recursos de KDE
+Comment[hne]=केडीई संसाधन मन ल कान्फिगर करव
 Comment[hu]=A KDE-erőforrások beállítása
 Comment[it]=Configura le risorse di KDE
 Comment[ja]=KDE リソースの設定
Index: kresources/kresources_plugin.desktop
===================================================================
--- kresources/kresources_plugin.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kresources/kresources_plugin.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -10,7 +10,9 @@
 Name[fr]=Module externe de gestion de ressources
 Name[ga]=Breiseán KResources
 Name[gl]=Extensión de KResources
+Name[hne]=केरिसोर्स प्लगइन
 Name[hu]=KResources-modul
+Name[is]=KResources íforrit
 Name[it]=Plugin di KResources
 Name[ja]=KDE リソースプラグイン
 Name[km]=កម្មវិធី​ជំនួយ​របស់ KResources
@@ -50,7 +52,9 @@
 Comment[fr]=Module externe d'infrastructure de ressources
 Comment[ga]=Breiseán Creatlaí KResources
 Comment[gl]=Extensión da infraestrutura KResource
+Comment[hne]=केरिसोर्स फ्रेमवर्क प्लगइन
 Comment[hu]=Modul a KResources keretrendszerhez
+Comment[is]=KResources grunníforrit
 Comment[it]=Plugin di infrastruttura di KResources
 Comment[ja]=KDE リソースフレームワークのプラグイン
 Comment[km]=កម្មវិធី​ជំនួយ​គ្រោងការ​របស់ KResources
Index: doc/kcontrol/kresources/index.docbook
===================================================================
--- doc/kcontrol/kresources/index.docbook	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ doc/kcontrol/kresources/index.docbook	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -39,7 +39,7 @@
 
 <para>Use the buttons on the right side of the listbox to add, remove or edit resources.</para>
 
-<para>Detailed informations about resource types and options you find in the links provided in the following table:</para>
+<para>Detailed information about resource types and options you find in the links provided in the following table:</para>
 
 <para>
 <table>
Index: kcal/resourcelocaldir.cpp
===================================================================
--- kcal/resourcelocaldir.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/resourcelocaldir.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -142,13 +142,13 @@
   const QStringList entries = dir.entryList( QDir::Files | QDir::Readable );
 
   bool success = true;
-  QStringList::ConstIterator it;
-  for ( it = entries.begin(); it != entries.end(); ++it ) {
-    if ( (*it).endsWith( '~' ) ) { // is backup file, ignore it
-      continue;
+  for ( int i = 0, count = entries.count(); i < count; ++i ) {
+    if ( entries[i].contains( QRegExp( "(~|\\.new|\\.tmp)$" ) ) ||
+         entries[i].startsWith( d->mURL.path() + "/qt_temp." ) ) {
+      continue;  // backup or temporary file, ignore it
     }
 
-    const QString fileName = dirName + '/' + *it;
+    const QString fileName = dirName + '/' + entries[i];
     kDebug() << " read '" << fileName << "'";
     CalendarLocal cal( calendar()->timeSpec() );
     if ( !doFileLoad( cal, fileName ) ) {
@@ -195,6 +195,11 @@
 
 bool ResourceLocalDir::doSave( bool, Incidence *incidence )
 {
+  if ( d->mDeletedIncidences.contains( incidence ) ) {
+    d->mDeletedIncidences.removeAll( incidence );
+    return true;
+  }
+
   d->mDirWatch.stopScan();  // do prohibit the dirty() signal and a following reload()
 
   QString fileName = d->mURL.path() + '/' + incidence->uid();
@@ -218,7 +223,9 @@
 {
   kDebug();
 
-  if ( !isOpen() ) {
+  if ( !isOpen() ||
+       file.contains( QRegExp( "(~|\\.new|\\.tmp)$" ) ) ||
+       file.startsWith( d->mURL.path() + "/qt_temp." ) ) {
     return;
   }
 
@@ -234,7 +241,12 @@
 {
   kDebug();
   if ( d->deleteIncidenceFile( event ) ) {
-    return calendar()->deleteEvent( event );
+    if ( calendar()->deleteEvent( event ) ) {
+      d->mDeletedIncidences.append( event );
+      return true;
+    } else {
+      return false;
+    }
   } else {
     return false;
   }
@@ -248,7 +260,12 @@
 bool ResourceLocalDir::deleteTodo( Todo *todo )
 {
   if ( d->deleteIncidenceFile( todo ) ) {
-    return calendar()->deleteTodo( todo );
+    if ( calendar()->deleteTodo( todo ) ) {
+      d->mDeletedIncidences.append( todo );
+      return true;
+    } else {
+      return false;
+    }
   } else {
     return false;
   }
@@ -262,7 +279,12 @@
 bool ResourceLocalDir::deleteJournal( Journal *journal )
 {
   if ( d->deleteIncidenceFile( journal ) ) {
-    return calendar()->deleteJournal( journal );
+    if ( calendar()->deleteJournal( journal ) ) {
+      d->mDeletedIncidences.append( journal );
+      return true;
+    } else {
+      return false;
+    }
   } else {
     return false;
   }
Index: kcal/localdir.desktop
===================================================================
--- kcal/localdir.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/localdir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[fr]=Calendrier dans un dossier local
 Name[ga]=Féilire i gComhadlann Logánta
 Name[gl]=Axenda nun cartafol local
+Name[hne]=लोकल डिरेक्टरी मं कलेन्डर
 Name[hu]=Naptár helyi könyvtárban
 Name[it]=Calendario nella cartella locale
 Name[ja]=ローカルディレクトリのカレンダー
Index: kcal/local.desktop
===================================================================
--- kcal/local.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/local.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[fr]=Calendrier dans un fichier local
 Name[ga]=Féilire i gComhad Logánta
 Name[gl]=Axenda nun ficheiro local
+Name[hne]=लोकल फाइल मं कलेन्डर
 Name[hu]=Helyi naptárfájl
 Name[it]=Calendario in file locale
 Name[ja]=ローカルファイルのカレンダー
Index: kcal/kcal_manager.desktop
===================================================================
--- kcal/kcal_manager.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/kcal_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -12,6 +12,7 @@
 Name[fr]=Calendrier
 Name[ga]=Féilire
 Name[gl]=Axenda
+Name[hne]=कलेन्डर
 Name[hu]=Naptár
 Name[it]=Calendario
 Name[ja]=カレンダー
Index: kcal/vcalformat.cpp
===================================================================
--- kcal/vcalformat.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/vcalformat.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -263,8 +263,10 @@
 
   // organizer stuff
   // @TODO: How about the common name?
-  tmpStr = "MAILTO:" + anEvent->organizer().email();
-  addPropValue( vtodo, ICOrganizerProp, tmpStr.toLocal8Bit() );
+  if ( !anEvent->organizer().email().isEmpty() ) {
+    tmpStr = "MAILTO:" + anEvent->organizer().email();
+    addPropValue( vtodo, ICOrganizerProp, tmpStr.toLocal8Bit() );
+  }
 
   // attendees
   if ( anEvent->attendeeCount() > 0 ) {
@@ -421,8 +423,10 @@
 
   // attendee and organizer stuff
   // TODO: What to do with the common name?
-  tmpStr = "MAILTO:" + anEvent->organizer().email();
-  addPropValue( vevent, ICOrganizerProp, tmpStr.toLocal8Bit() );
+  if ( !anEvent->organizer().email().isEmpty() ) {
+    tmpStr = "MAILTO:" + anEvent->organizer().email();
+    addPropValue( vevent, ICOrganizerProp, tmpStr.toLocal8Bit() );
+  }
 
   // TODO: Put this functionality into Attendee class
   if ( anEvent->attendeeCount() > 0 ) {
@@ -553,7 +557,7 @@
   }
   if ( !tmpStr2.isEmpty() ) {
     tmpStr2.truncate( tmpStr2.length() - 1 );
-    addPropValue( vevent, VCExDateProp, tmpStr2.toLocal8Bit() );
+    addPropValue( vevent, VCExpDateProp, tmpStr2.toLocal8Bit() );
   }
 
   // description
@@ -1197,7 +1201,7 @@
   } // repeats
 
   // recurrence exceptions
-  if ( ( vo = isAPropertyOf( vevent, VCExDateProp ) ) != 0 ) {
+  if ( ( vo = isAPropertyOf( vevent, VCExpDateProp ) ) != 0 ) {
     s = fakeCString( vObjectUStringZValue( vo ) );
     QStringList exDates = QString::fromLocal8Bit( s ).split( ',' );
     QStringList::ConstIterator it;
Index: kcal/icalformat_p.cpp
===================================================================
--- kcal/icalformat_p.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/icalformat_p.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -580,8 +580,10 @@
 
   // organizer stuff
   if ( !incidenceBase->organizer().isEmpty() ) {
-    icalcomponent_add_property(
-      parent, mImpl->writeOrganizer( incidenceBase->organizer() ) );
+    icalproperty *p = mImpl->writeOrganizer( incidenceBase->organizer() );
+    if ( p ) {
+      icalcomponent_add_property( parent, p );
+    }
   }
 
   // attendees
@@ -589,7 +591,10 @@
     Attendee::List::ConstIterator it;
     for ( it = incidenceBase->attendees().constBegin();
           it != incidenceBase->attendees().constEnd(); ++it ) {
-      icalcomponent_add_property( parent, mImpl->writeAttendee( *it ) );
+      icalproperty *p = mImpl->writeAttendee( *it );
+      if ( p ) {
+        icalcomponent_add_property( parent, p );
+      }
     }
   }
 
@@ -617,9 +622,12 @@
 
 icalproperty *ICalFormatImpl::writeOrganizer( const Person &organizer )
 {
-  icalproperty *p =
-    icalproperty_new_organizer( "MAILTO:" + organizer.email().toUtf8() );
+  if ( organizer.email().isEmpty() ) {
+    return 0;
+  }
 
+  icalproperty *p = icalproperty_new_organizer( "MAILTO:" + organizer.email().toUtf8() );
+
   if ( !organizer.name().isEmpty() ) {
     icalproperty_add_parameter(
       p, icalparameter_new_cn( quoteForParam( organizer.name() ).toUtf8() ) );
@@ -658,6 +666,10 @@
 
 icalproperty *ICalFormatImpl::writeAttendee( Attendee *attendee )
 {
+  if ( attendee->email().isEmpty() ) {
+    return 0;
+  }
+
   icalproperty *p =
     icalproperty_new_attendee( "mailto:" + attendee->email().toUtf8() );
 
@@ -937,13 +949,14 @@
     const QList<Person> addresses = alarm->mailAddresses();
     for ( QList<Person>::ConstIterator ad = addresses.constBegin();
           ad != addresses.constEnd();  ++ad ) {
-      icalproperty *p = icalproperty_new_attendee(
-        "MAILTO:" + (*ad).email().toUtf8() );
-      if ( !(*ad).name().isEmpty() ) {
-        icalproperty_add_parameter(
-          p, icalparameter_new_cn( quoteForParam( (*ad).name() ).toUtf8() ) );
+      if ( !(*ad).email().isEmpty() ) {
+        icalproperty *p = icalproperty_new_attendee( "MAILTO:" + (*ad).email().toUtf8() );
+        if ( !(*ad).name().isEmpty() ) {
+          icalproperty_add_parameter(
+            p, icalparameter_new_cn( quoteForParam( (*ad).name() ).toUtf8() ) );
+        }
+        icalcomponent_add_property( a, p );
       }
-      icalcomponent_add_property( a, p );
     }
     icalcomponent_add_property(
       a, icalproperty_new_summary( alarm->mailSubject().toUtf8() ) );
Index: kcal/calendarresources.cpp
===================================================================
--- kcal/calendarresources.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/calendarresources.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -66,7 +66,8 @@
         mStandardPolicy( new StandardDestinationPolicy( mManager ) ),
         mDestinationPolicy( mStandardPolicy ),
         mAskPolicy( new AskDestinationPolicy( mManager ) ),
-        mException( 0 )
+        mException( 0 ),
+        mPendingDeleteFromResourceMap( false )
     {}
     ~Private()
     {
@@ -88,6 +89,8 @@
 
     ErrorFormat *mException;
 
+    bool mPendingDeleteFromResourceMap;
+
     template< class IncidenceList >
     void appendIncidences( IncidenceList &result, const IncidenceList &extra,
                            ResourceCalendar * );
@@ -212,7 +215,6 @@
   : Calendar( timeSpec ),
     d( new KCal::CalendarResources::Private( family ) )
 {
-  mPendingDeleteFromResourceMap = false;
   d->mManager->addObserver( this );
 }
 
@@ -221,7 +223,6 @@
   : Calendar( timeZoneId ),
     d( new KCal::CalendarResources::Private( family ) )
 {
-  mPendingDeleteFromResourceMap = false;
   d->mManager->addObserver( this );
 }
 
@@ -442,7 +443,7 @@
   if ( d->mResourceMap.find( event ) != d->mResourceMap.end() ) {
     status = d->mResourceMap[event]->deleteEvent( event );
     if ( status ) {
-      mPendingDeleteFromResourceMap = true;
+      d->mPendingDeleteFromResourceMap = true;
     }
   } else {
     status = false;
@@ -498,7 +499,7 @@
   if ( d->mResourceMap.find( todo ) != d->mResourceMap.end() ) {
     status = d->mResourceMap[todo]->deleteTodo( todo );
     if ( status ) {
-      mPendingDeleteFromResourceMap = true;
+      d->mPendingDeleteFromResourceMap = true;
     }
   } else {
     CalendarResourceManager::ActiveIterator it;
@@ -644,7 +645,7 @@
   if ( d->mResourceMap.find( journal ) != d->mResourceMap.end() ) {
     status = d->mResourceMap[journal]->deleteJournal( journal );
     if ( status ) {
-      mPendingDeleteFromResourceMap = true;
+      d->mPendingDeleteFromResourceMap = true;
     }
   } else {
     CalendarResourceManager::ActiveIterator it;
@@ -845,7 +846,7 @@
     }
     d->mResourceMap[ incidence ] = r;
   }
-  mPendingDeleteFromResourceMap = false;
+  d->mPendingDeleteFromResourceMap = false;
 
   int count = incrementChangeCount( r );
   if ( count == 1 ) {
@@ -871,9 +872,9 @@
 
   int count = decrementChangeCount( r );
 
-  if ( mPendingDeleteFromResourceMap ) {
+  if ( d->mPendingDeleteFromResourceMap ) {
     d->mResourceMap.remove( incidence );
-    mPendingDeleteFromResourceMap = false;
+    d->mPendingDeleteFromResourceMap = false;
   }
 
   if ( count == 0 ) {
Index: kcal/calendarresources.h
===================================================================
--- kcal/calendarresources.h	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/calendarresources.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -721,7 +721,6 @@
     Q_DISABLE_COPY( CalendarResources )
     class Private;
     Private *d;
-    bool mPendingDeleteFromResourceMap;
     //@endcond
 };
 
Index: kcal/tests/testtodo.cpp
===================================================================
--- kcal/tests/testtodo.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/testtodo.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -98,3 +98,24 @@
   Todo todo2 = todo1;
   QVERIFY( todo1 == todo2 );
 }
+
+void TodoTest::testSetCompleted() {
+
+  Todo todo1, todo2;
+  todo1.setSummary( "Todo Summary" );
+  todo2.setSummary( "Todo Summary" );
+  KDateTime today = KDateTime::currentUtcDateTime();
+
+  // due yesterday
+  KDateTime originalDueDate = today.addDays( -1 );
+ 
+  todo1.setDtDue( originalDueDate );
+  todo1.recurrence()->setDaily( 1 );
+  todo1.setCompleted( today );
+
+  todo2.setCompleted( true );
+
+  QVERIFY( originalDueDate != todo1.dtDue() );
+  QVERIFY( !todo1.isCompleted() );
+  QVERIFY( todo2.isCompleted() );
+}
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase12.ics
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase12.ics	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase12.ics	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -12,7 +12,7 @@
 SUMMARY:20. Mai 2005\, 13-14 Uhr\, alle 3 Tage\, bis 30. Juni 2005
 CLASS:PUBLIC
 PRIORITY:5
-RRULE:FREQ=DAILY;UNTIL=20050630;INTERVAL=3
+RRULE:FREQ=DAILY;UNTIL=20050630T123400Z;INTERVAL=3
 DTSTART:20050520T110000Z
 DTEND:20050520T120000Z
 TRANSP:OPAQUE
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase13.ics.vcal.ref
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase13.ics.vcal.ref	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase13.ics.vcal.ref	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -9,7 +9,7 @@
 SEQUENCE:1
 LAST-MODIFIED:20050520T105327Z
 X-ORGANIZER:MAILTO:reinhold@kainhofer.com
-RRULE:MP3 1- TH 20061231T050000
+RRULE:MP3 1- TH 20061231T123400
 SUMMARY:20. Mai 2005, 20-21 Uhr, alle 3 Monate am letzten Do, bis 31. 12. 2006
 CLASS:PUBLIC
 PRIORITY:5
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase05.ics.vcal.ref
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase05.ics.vcal.ref	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase05.ics.vcal.ref	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -9,7 +9,7 @@
 SEQUENCE:2
 LAST-MODIFIED:20050520T105740Z
 X-ORGANIZER:MAILTO:reinhold@kainhofer.com
-RRULE:W2 MO WE FR 20050630T040000
+RRULE:W2 MO WE FR 20050630T123400
 SUMMARY:20. Mai 2005, 14-15 Uhr, alle 2 Wochen am Mo/Mi/Fr, bis 30. Juni 2005
 CLASS:PUBLIC
 PRIORITY:5
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase13.ics
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase13.ics	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase13.ics	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -13,7 +13,7 @@
   12. 2006
 CLASS:PUBLIC
 PRIORITY:5
-RRULE:FREQ=MONTHLY;UNTIL=20061231;INTERVAL=3;BYDAY=-1TH
+RRULE:FREQ=MONTHLY;UNTIL=20061231T123400Z;INTERVAL=3;BYDAY=-1TH
 DTSTART:20050520T180000Z
 DTEND:20050520T190000Z
 TRANSP:OPAQUE
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase12.ics.vcal.ref
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase12.ics.vcal.ref	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase12.ics.vcal.ref	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -9,7 +9,7 @@
 SEQUENCE:1
 LAST-MODIFIED:20050520T104819Z
 X-ORGANIZER:MAILTO:reinhold@kainhofer.com
-RRULE:D3 20050630T040000
+RRULE:D3 20050630T123400
 SUMMARY:20. Mai 2005, 13-14 Uhr, alle 3 Tage, bis 30. Juni 2005
 CLASS:PUBLIC
 PRIORITY:5
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase04.ics.vcal.ref
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase04.ics.vcal.ref	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase04.ics.vcal.ref	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -9,7 +9,7 @@
 SEQUENCE:4
 LAST-MODIFIED:20050520T105750Z
 X-ORGANIZER:MAILTO:reinhold@kainhofer.com
-RRULE:MD3 20 20060630T040000
+RRULE:MD3 20 20060630T123400
 SUMMARY:20. Mai 2005, 16-17 Uhr, alle 3 Monate am 20., bis 30. Juni 2006
 CLASS:PUBLIC
 PRIORITY:5
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase04.ics
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase04.ics	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase04.ics	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -13,7 +13,7 @@
   2006
 CLASS:PUBLIC
 PRIORITY:5
-RRULE:FREQ=MONTHLY;UNTIL=20060630;INTERVAL=3;BYMONTHDAY=20
+RRULE:FREQ=MONTHLY;UNTIL=20060630T123400Z;INTERVAL=3;BYMONTHDAY=20
 DTSTART:20050520T140000Z
 DTEND:20050520T150000Z
 TRANSP:OPAQUE
Index: kcal/tests/data/vCalendar/KOrganizer_vCalTestCase05.ics
===================================================================
--- kcal/tests/data/vCalendar/KOrganizer_vCalTestCase05.ics	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/data/vCalendar/KOrganizer_vCalTestCase05.ics	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -13,7 +13,7 @@
   Juni 2005
 CLASS:PUBLIC
 PRIORITY:5
-RRULE:FREQ=WEEKLY;UNTIL=20050630;INTERVAL=2;BYDAY=MO,WE,FR
+RRULE:FREQ=WEEKLY;UNTIL=20050630T123400Z;INTERVAL=2;BYDAY=MO,WE,FR
 DTSTART:20050520T120000Z
 DTEND:20050520T130000Z
 TRANSP:OPAQUE
Index: kcal/tests/testtodo.h
===================================================================
--- kcal/tests/testtodo.h	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/tests/testtodo.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -31,6 +31,7 @@
     void testCompare();
     void testClone();
     void testAssign();
+    void testSetCompleted();
 };
 
 #endif
Index: kcal/resourcelocaldir_p.h
===================================================================
--- kcal/resourcelocaldir_p.h	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/resourcelocaldir_p.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -56,6 +56,7 @@
     KABC::Lock *mLock;
     KUrl mURL;
     KDirWatch mDirWatch;
+    QList<Incidence *> mDeletedIncidences;
 };
 //@endcond
 
Index: kcal/CMakeLists.txt
===================================================================
--- kcal/CMakeLists.txt	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/CMakeLists.txt	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -1,10 +1,7 @@
 project(kcal)
 
-macro_optional_find_package(Libical)
+find_package(Libical)
 macro_log_feature(LIBICAL_FOUND "libical" "Reference implementation of the iCalendar data type and serialization format" "http://sourceforge.net/projects/freeassociation" TRUE "0.33" "Required for the critical PIM kcal library.")
-if(NOT LIBICAL_FOUND)
-  macro_display_feature_log()
-endif(NOT LIBICAL_FOUND)
 
 add_definitions(-DKDE_DEFAULT_DEBUG_AREA=5800)
 
Index: kcal/todo.cpp
===================================================================
--- kcal/todo.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/todo.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -508,7 +508,9 @@
       while ( !todo->recursAt( nextDate ) ||
               nextDate <= KDateTime::currentUtcDateTime() ) {
 
-        if ( !nextDate.isValid() || nextDate > endDateTime ) {
+        if ( !nextDate.isValid() ||
+	        ( nextDate > endDateTime && r->duration() != -1 ) ) {
+
           return false;
         }
 
Index: kcal/versit/vobject.h
===================================================================
--- kcal/versit/vobject.h	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/versit/vobject.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -142,7 +142,7 @@
 #define VCEventProp				"VEVENT"
 #define VCEWorldProp			"EWORLD"
 #define VCExNumProp				"EXNUM"
-#define VCExDateProp			"EXDATE"
+#define VCExpDateProp			"EXDATE"
 #define VCExpectProp			"EXPECT"
 #define VCExtAddressProp		"EXT ADD"
 #define VCFamilyNameProp		"F"
Index: kcal/versit/vobject.c
===================================================================
--- kcal/versit/vobject.c	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/versit/vobject.c	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -804,7 +804,7 @@
     { VCEventProp, 0, 0, PD_BEGIN },
     { VCEWorldProp, 0, 0, 0 },
     { VCExNumProp, 0, 0, 0 },
-    { VCExDateProp, 0, 0, 0 },
+    { VCExpDateProp, 0, 0, 0 },
     { VCExpectProp, 0, 0, 0 },
     { VCExtAddressProp, 0, 0, 0 },
     { VCFamilyNameProp, 0, 0, 0 },
Index: kcal/alarm.cpp
===================================================================
--- kcal/alarm.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kcal/alarm.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -412,11 +412,13 @@
   if ( hasTime() ) {
     return d->mAlarmTime;
   } else if ( d->mParent ) {
-    if ( d->mParent->type() == "Todo" ) {
-      Todo *t = static_cast<Todo*>( d->mParent );
-      return d->mOffset.end( t->dtDue() );
-    } else if ( d->mEndOffset ) {
-      return d->mOffset.end( d->mParent->dtEnd() );
+    if ( d->mEndOffset ) {
+      if ( d->mParent->type() == "Todo" ) {
+        Todo *t = static_cast<Todo*>( d->mParent );
+        return d->mOffset.end( t->dtDue() );
+      } else {
+        return d->mOffset.end( d->mParent->dtEnd() );
+      }
     } else {
       return d->mOffset.end( d->mParent->dtStart() );
     }
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ CMakeLists.txt	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -5,11 +5,11 @@
 # Used e.g. in KdepimLibsConfig.cmake, Alex
 set(KDEPIMLIBS_VERSION_MAJOR 4)
 set(KDEPIMLIBS_VERSION_MINOR 2)
-set(KDEPIMLIBS_VERSION_PATCH 0)
+set(KDEPIMLIBS_VERSION_PATCH 1)
 set(KDEPIMLIBS_VERSION ${KDEPIMLIBS_VERSION_MAJOR}.${KDEPIMLIBS_VERSION_MINOR}.${KDEPIMLIBS_VERSION_PATCH} )
 
 # search packages used by KDE
-find_package(KDE4  4.2.0  REQUIRED)
+find_package(KDE4  4.2.1  REQUIRED)
 
 include (KDE4Defaults)
 include (MacroLibrary)
Index: kioslave/imap4/imapparser.cpp
===================================================================
--- kioslave/imap4/imapparser.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/imap4/imapparser.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -80,8 +80,8 @@
   lastHandled = 0;
 }
 
-imapCommand *
-imapParser::doCommand (imapCommand * aCmd)
+CommandPtr
+imapParser::doCommand (CommandPtr aCmd)
 {
   int pl = 0;
   sendCommand (aCmd);
@@ -93,8 +93,8 @@
   return aCmd;
 }
 
-imapCommand *
-imapParser::sendCommand (imapCommand * aCmd)
+CommandPtr
+imapParser::sendCommand (CommandPtr aCmd)
 {
   aCmd->setId (QString::number(commandCounter++));
   sentQueue.append (aCmd);
@@ -142,13 +142,13 @@
 imapParser::clientLogin (const QString & aUser, const QString & aPass,
   QString & resultInfo)
 {
-  imapCommand *cmd;
+  CommandPtr cmd;
   bool retVal = false;
 
   cmd =
-    doCommand (new
-               imapCommand ("LOGIN", "\"" + KIMAP::quoteIMAP(aUser)
-               + "\" \"" + KIMAP::quoteIMAP(aPass) + "\""));
+    doCommand ( CommandPtr( new
+                            imapCommand ("LOGIN", "\"" + KIMAP::quoteIMAP(aUser)
+                                         + "\" \"" + KIMAP::quoteIMAP(aPass) + "\"")) );
 
   if (cmd->result () == "OK")
   {
@@ -157,7 +157,6 @@
   }
   resultInfo = cmd->resultInfo();
   completeQueue.removeAll (cmd);
-  delete cmd;
   return retVal;
 }
 
@@ -257,7 +256,7 @@
     sasl_dispose( &conn );
     return false;
   }
-  imapCommand *cmd;
+  CommandPtr cmd;
 
   tmp = QByteArray::fromRawData( out, outlen );
   challenge = tmp.toBase64();
@@ -268,7 +267,7 @@
     firstCommand += ' ';
     firstCommand += QString::fromLatin1( challenge.data(), challenge.size() );
   }
-  cmd = sendCommand (new imapCommand ("AUTHENTICATE", firstCommand.toLatin1()));
+  cmd = sendCommand (CommandPtr(new imapCommand ("AUTHENTICATE", firstCommand.toLatin1())));
 
   while ( true )
   {
@@ -353,7 +352,7 @@
       parseResult (what, result);
       if ( sentQueue.count() ) {
         // BYE that interrupts a command -> copy the reason for it
-        imapCommand *current = sentQueue.at (0);
+        CommandPtr current = sentQueue.at (0);
         current->setResultInfo(result.cstr());
       }
       currentState = ISTATE_NO;
@@ -1745,7 +1744,7 @@
   }
   else
   {
-    imapCommand *current = sentQueue.at (0);
+    CommandPtr current = sentQueue.at (0);
     switch (result[0])
     {
     case '*':
Index: kioslave/imap4/imapparser.h
===================================================================
--- kioslave/imap4/imapparser.h	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/imap4/imapparser.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -205,13 +205,13 @@
    * @param aCmd The command to perform
    * @return The completed command
    */
-  imapCommand *sendCommand (imapCommand * aCmd);
+  CommandPtr sendCommand (CommandPtr aCmd);
   /**
    * @brief perform a command and wait to parse the result
    * @param aCmd The command to perform
    * @return The completed command
    */
-  imapCommand *doCommand (imapCommand * aCmd);
+  CommandPtr doCommand (CommandPtr aCmd);
 
 
   /**
@@ -435,8 +435,8 @@
   QList < imapList > listResponses;
 
   /** @brief queues handling the running commands */
-  QList < imapCommand *> sentQueue;  // no autodelete
-  QList < imapCommand *> completeQueue;  // autodelete !!
+  QList <CommandPtr> sentQueue;
+  QList <CommandPtr> completeQueue;
 
   /**
    * everything we didn't handle, everything but the greeting is bogus
Index: kioslave/imap4/imap4.cpp
===================================================================
--- kioslave/imap4/imap4.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/imap4/imap4.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -206,7 +206,7 @@
 
   if (aSequence == "0:0" && getState() == ISTATE_SELECT)
   {
-    imapCommand *cmd = doCommand (imapCommand::clientNoop());
+    CommandPtr cmd = doCommand (imapCommand::clientNoop());
     completeQueue.removeAll(cmd);
   }
 
@@ -216,7 +216,7 @@
   }
 
   mProcessedSize = 0;
-  imapCommand *cmd = NULL;
+  CommandPtr cmd;
   if (!assureBox (aBox, true)) return;
 
 #ifdef USE_VALIDITY
@@ -428,7 +428,7 @@
   if (myType == ITYPE_DIR || myType == ITYPE_DIR_AND_BOX)
   {
     QString listStr = myBox;
-    imapCommand *cmd;
+    CommandPtr cmd;
 
     if (!listStr.isEmpty () && !listStr.endsWith(myDelimiter) &&
         mySection != "FOLDERONLY")
@@ -513,7 +513,7 @@
       query = query.right (query.length () - 1);
       if (!query.isEmpty())
       {
-        imapCommand *cmd = NULL;
+        CommandPtr cmd;
 
         if (!assureBox (myBox, true)) return;
 
@@ -598,7 +598,7 @@
         if (mySection.isEmpty()) mySection = "UID RFC822.SIZE ENVELOPE";
 
         bool withFlags = mySection.toUpper().contains("FLAGS") ;
-        imapCommand *fetch =
+        CommandPtr fetch =
           sendCommand (imapCommand::
                        clientFetch (mySequence, mySection));
         imapCache *cache;
@@ -796,7 +796,7 @@
   {
     if (aBox[aBox.length () - 1] == '/')
       aBox = aBox.right (aBox.length () - 1);
-    imapCommand *cmd = doCommand (imapCommand::clientCreate (aBox));
+    CommandPtr cmd = doCommand (imapCommand::clientCreate (aBox));
 
     if (cmd->result () != "OK") {
       error (ERR_COULD_NOT_WRITE, _url.prettyUrl());
@@ -833,7 +833,7 @@
       return;
     }
 
-    imapCommand *cmd =
+    CommandPtr cmd =
       sendCommand (imapCommand::clientAppend (aBox, aSection, length));
     while (!parseLoop ()) {}
 
@@ -926,7 +926,7 @@
   QString aBox, aSequence, aLType, aSection, aValidity, aDelimiter, aInfo;
   parseURL(_url, aBox, aSection, aLType, aSequence, aValidity, aDelimiter, aInfo);
   kDebug(7116) <<"IMAP4::mkdir - create" << aBox;
-  imapCommand *cmd = doCommand (imapCommand::clientCreate(aBox));
+  CommandPtr cmd = doCommand (imapCommand::clientCreate(aBox));
 
   if (cmd->result () != "OK")
   {
@@ -1018,7 +1018,7 @@
         if (dType != ITYPE_BOX && dType != ITYPE_DIR_AND_BOX)
         {
           // ok then we'll create a mailbox
-          imapCommand *cmd = doCommand (imapCommand::clientCreate (topDir));
+          CommandPtr cmd = doCommand (imapCommand::clientCreate (topDir));
 
           // on success we'll use it, else we'll just try to create the given dir
           if (cmd->result () == "OK")
@@ -1049,7 +1049,7 @@
     kDebug(7116) <<"IMAP4::copy -" << sBox <<" ->" << dBox;
 
     //issue copy command
-    imapCommand *cmd =
+    CommandPtr cmd =
       doCommand (imapCommand::clientCopy (dBox, sSequence));
     if (cmd->result () != "OK")
     {
@@ -1096,7 +1096,7 @@
       if (aSequence == "*")
       {
         if (!assureBox (aBox, false)) return;
-        imapCommand *cmd = doCommand (imapCommand::clientExpunge ());
+        CommandPtr cmd = doCommand (imapCommand::clientExpunge ());
         if (cmd->result () != "OK") {
           error (ERR_CANNOT_DELETE, _url.prettyUrl());
           completeQueue.removeAll (cmd);
@@ -1108,7 +1108,7 @@
       {
         // if open for read/write
         if (!assureBox (aBox, false)) return;
-        imapCommand *cmd =
+        CommandPtr cmd =
           doCommand (imapCommand::
                      clientStore (aSequence, "+FLAGS.SILENT", "\\DELETED"));
         if (cmd->result () != "OK") {
@@ -1123,12 +1123,12 @@
     {
       if (getCurrentBox() == aBox)
       {
-        imapCommand *cmd = doCommand(imapCommand::clientClose());
+        CommandPtr cmd = doCommand(imapCommand::clientClose());
         completeQueue.removeAll(cmd);
         setState(ISTATE_LOGIN);
       }
       // We unsubscribe, otherwise we get ghost folders on UW-IMAP
-      imapCommand *cmd = doCommand(imapCommand::clientUnsubscribe(aBox));
+      CommandPtr cmd = doCommand(imapCommand::clientUnsubscribe(aBox));
       completeQueue.removeAll(cmd);
       cmd = doCommand(imapCommand::clientDelete (aBox));
       // If this doesn't work, we try to empty the mailbox first
@@ -1139,21 +1139,21 @@
         bool stillOk = true;
         if (stillOk)
         {
-          imapCommand *cmd = doCommand(
+          CommandPtr cmd = doCommand(
             imapCommand::clientStore("1:*", "+FLAGS.SILENT", "\\DELETED"));
           if (cmd->result () != "OK") stillOk = false;
           completeQueue.removeAll(cmd);
         }
         if (stillOk)
         {
-          imapCommand *cmd = doCommand(imapCommand::clientClose());
+          CommandPtr cmd = doCommand(imapCommand::clientClose());
           if (cmd->result () != "OK") stillOk = false;
           completeQueue.removeAll(cmd);
           setState(ISTATE_LOGIN);
         }
         if (stillOk)
         {
-          imapCommand *cmd = doCommand (imapCommand::clientDelete(aBox));
+          CommandPtr cmd = doCommand (imapCommand::clientDelete(aBox));
           if (cmd->result () != "OK") stillOk = false;
           completeQueue.removeAll(cmd);
         }
@@ -1170,7 +1170,7 @@
 
   case ITYPE_DIR:
     {
-      imapCommand *cmd = doCommand (imapCommand::clientDelete (aBox));
+      CommandPtr cmd = doCommand (imapCommand::clientDelete (aBox));
       if (cmd->result () != "OK") {
         error (ERR_COULD_NOT_RMDIR, _url.prettyUrl());
         completeQueue.removeAll (cmd);
@@ -1184,7 +1184,7 @@
     {
       // if open for read/write
       if (!assureBox (aBox, false)) return;
-      imapCommand *cmd =
+      CommandPtr cmd =
         doCommand (imapCommand::
                    clientStore (aSequence, "+FLAGS.SILENT", "\\DELETED"));
       if (cmd->result () != "OK") {
@@ -1251,7 +1251,7 @@
   case 'N':
   {
     // NOOP
-    imapCommand *cmd = doCommand(imapCommand::clientNoop());
+    CommandPtr cmd = doCommand(imapCommand::clientNoop());
     if (cmd->result () != "OK")
     {
       kDebug(7116) <<"NOOP did not succeed - connection broken";
@@ -1260,7 +1260,6 @@
       return;
     }
     completeQueue.removeAll (cmd);
-    delete cmd;
     finished();
     break;
   }
@@ -1279,7 +1278,7 @@
     stream >> _url;
     QString aBox, aSequence, aLType, aSection, aValidity, aDelimiter, aInfo;
     parseURL (_url, aBox, aSection, aLType, aSequence, aValidity, aDelimiter, aInfo);
-    imapCommand *cmd = doCommand(imapCommand::clientUnsubscribe(aBox));
+    CommandPtr cmd = doCommand(imapCommand::clientUnsubscribe(aBox));
     if (cmd->result () != "OK")
     {
       completeQueue.removeAll (cmd);
@@ -1300,7 +1299,7 @@
     stream >> _url;
     QString aBox, aSequence, aLType, aSection, aValidity, aDelimiter, aInfo;
     parseURL (_url, aBox, aSection, aLType, aSequence, aValidity, aDelimiter, aInfo);
-    imapCommand *cmd = doCommand(imapCommand::clientSubscribe(aBox));
+    CommandPtr cmd = doCommand(imapCommand::clientSubscribe(aBox));
     if (cmd->result () != "OK")
     {
       completeQueue.removeAll (cmd);
@@ -1368,8 +1367,8 @@
       knownFlags += " KMAILFORWARDED KMAILTODO KMAILWATCHED KMAILIGNORED $FORWARDED $TODO $WATCHED $IGNORED";
     }
 
-    imapCommand *cmd = doCommand (imapCommand::
-                                  clientStore (aSequence, "-FLAGS.SILENT", knownFlags));
+    CommandPtr cmd = doCommand (imapCommand::
+                                clientStore (aSequence, "-FLAGS.SILENT", knownFlags));
     if (cmd->result () != "OK")
     {
       completeQueue.removeAll (cmd);
@@ -1407,7 +1406,7 @@
     if ( !assureBox(aBox, true) ) // read-only because changing SEEN should be possible even then
       return;
 
-    imapCommand *cmd;
+    CommandPtr cmd;
     if ( seen )
       cmd = doCommand( imapCommand::clientStore( aSequence, "+FLAGS.SILENT", "\\SEEN" ) );
     else
@@ -1459,7 +1458,7 @@
     QString user, acl;
     stream >> user >> acl;
     kDebug(7116) <<"SETACL" << aBox << user << acl;
-    imapCommand *cmd = doCommand(imapCommand::clientSetACL(aBox, user, acl));
+    CommandPtr cmd = doCommand(imapCommand::clientSetACL(aBox, user, acl));
     if (cmd->result () != "OK")
     {
       error(ERR_SLAVE_DEFINED, i18n("Setting the Access Control List on folder %1 "
@@ -1478,7 +1477,7 @@
     QString user;
     stream >> user;
     kDebug(7116) <<"DELETEACL" << aBox << user;
-    imapCommand *cmd = doCommand(imapCommand::clientDeleteACL(aBox, user));
+    CommandPtr cmd = doCommand(imapCommand::clientDeleteACL(aBox, user));
     if (cmd->result () != "OK")
     {
       error(ERR_SLAVE_DEFINED, i18n("Deleting the Access Control List on folder %1 "
@@ -1495,7 +1494,7 @@
   case 'G': // GETACL
   {
     kDebug(7116) <<"GETACL" << aBox;
-    imapCommand *cmd = doCommand(imapCommand::clientGetACL(aBox));
+    CommandPtr cmd = doCommand(imapCommand::clientGetACL(aBox));
     if (cmd->result () != "OK")
     {
       error(ERR_SLAVE_DEFINED, i18n("Retrieving the Access Control List on folder %1 "
@@ -1522,7 +1521,7 @@
   case 'M': // MYRIGHTS
   {
     kDebug(7116) <<"MYRIGHTS" << aBox;
-    imapCommand *cmd = doCommand(imapCommand::clientMyRights(aBox));
+    CommandPtr cmd = doCommand(imapCommand::clientMyRights(aBox));
     if (cmd->result () != "OK")
     {
       error(ERR_SLAVE_DEFINED, i18n("Retrieving the Access Control List on folder %1 "
@@ -1556,7 +1555,7 @@
   parseURL (_url, aBox, aSection, aLType, aSequence, aValidity, aDelimiter, aInfo);
   if (!assureBox(aBox, false)) return;
 
-  imapCommand *cmd = doCommand (imapCommand::clientSearch( aSection ));
+  CommandPtr cmd = doCommand (imapCommand::clientSearch( aSection ));
   if (cmd->result () != "OK")
   {
     error(ERR_SLAVE_DEFINED, i18n("Searching of folder %1 "
@@ -1590,7 +1589,7 @@
    */
   if ( type == 'N' ) {
     kDebug(7116) << "IMAP4Protocol::specialCustomCommand: normal mode" << endl;
-    imapCommand *cmd = doCommand (imapCommand::clientCustom( command, arguments ));
+    CommandPtr cmd = doCommand (imapCommand::clientCustom( command, arguments ));
     if (cmd->result () != "OK")
     {
       error( ERR_SLAVE_DEFINED,
@@ -1613,7 +1612,7 @@
    */
   if ( type == 'E' ) {
     kDebug(7116) << "IMAP4Protocol::specialCustomCommand: extended mode" << endl;
-    imapCommand *cmd = sendCommand (imapCommand::clientCustom( command, QString() ));
+    CommandPtr cmd = sendCommand (imapCommand::clientCustom( command, QString() ));
     while ( !parseLoop () ) {};
 
     // see if server is waiting
@@ -1671,7 +1670,7 @@
     QMap<QString, QString> attributes;
     stream >> entry >> attributes;
     kDebug(7116) <<"SETANNOTATION" << aBox << entry << attributes.count() <<" attributes";
-    imapCommand *cmd = doCommand(imapCommand::clientSetAnnotation(aBox, entry, attributes));
+    CommandPtr cmd = doCommand(imapCommand::clientSetAnnotation(aBox, entry, attributes));
     if (cmd->result () != "OK")
     {
       error(ERR_SLAVE_DEFINED, i18n("Setting the annotation %1 on folder %2 "
@@ -1695,7 +1694,7 @@
     QStringList attributeNames;
     stream >> entry >> attributeNames;
     kDebug(7116) <<"GETANNOTATION" << aBox << entry << attributeNames;
-    imapCommand *cmd = doCommand(imapCommand::clientGetAnnotation(aBox, entry, attributeNames));
+    CommandPtr cmd = doCommand(imapCommand::clientGetAnnotation(aBox, entry, attributeNames));
     if (cmd->result () != "OK")
     {
       error(ERR_SLAVE_DEFINED, i18n("Retrieving the annotation %1 on folder %2 "
@@ -1732,7 +1731,7 @@
     case 'R': // GETQUOTAROOT
       {
         kDebug(7116) <<"QUOTAROOT" << aBox;
-        imapCommand *cmd = doCommand(imapCommand::clientGetQuotaroot( aBox ) );
+        CommandPtr cmd = doCommand(imapCommand::clientGetQuotaroot( aBox ) );
         if (cmd->result () != "OK")
         {
           error(ERR_SLAVE_DEFINED, i18n("Retrieving the quota root information on folder %1 "
@@ -1786,7 +1785,7 @@
         {
           kDebug(7116) <<"IMAP4::rename - close" << getCurrentBox();
           // mailbox can only be renamed if it is closed
-          imapCommand *cmd = doCommand (imapCommand::clientClose());
+          CommandPtr cmd = doCommand (imapCommand::clientClose());
           bool ok = cmd->result() == "OK";
           completeQueue.removeAll(cmd);
           if (!ok)
@@ -1796,7 +1795,7 @@
           }
           setState(ISTATE_LOGIN);
         }
-        imapCommand *cmd = doCommand (imapCommand::clientRename (sBox, dBox));
+        CommandPtr cmd = doCommand (imapCommand::clientRename (sBox, dBox));
         if (cmd->result () != "OK") {
           error (ERR_CANNOT_RENAME, cmd->result ());
           completeQueue.removeAll (cmd);
@@ -1854,7 +1853,7 @@
   {
     if (getState() == ISTATE_SELECT && aBox == getCurrentBox())
     {
-      imapCommand *cmd = doCommand (imapCommand::clientClose());
+      CommandPtr cmd = doCommand (imapCommand::clientClose());
       bool ok = cmd->result() == "OK";
       completeQueue.removeAll(cmd);
       if (!ok)
@@ -1870,7 +1869,7 @@
       ok = true;
     else
     {
-      imapCommand *cmd = doCommand(imapCommand::clientStatus(aBox, aSection));
+      CommandPtr cmd = doCommand(imapCommand::clientStatus(aBox, aSection));
       ok = cmd->result() == "OK";
       cmdInfo = cmd->resultInfo();
       completeQueue.removeAll(cmd);
@@ -1878,7 +1877,7 @@
     if (!ok)
     {
       bool found = false;
-      imapCommand *cmd = doCommand (imapCommand::clientList ("", aBox));
+      CommandPtr cmd = doCommand (imapCommand::clientList ("", aBox));
       if (cmd->result () == "OK")
       {
         for (QList< imapList >::Iterator it = listResponses.begin ();
@@ -1913,7 +1912,7 @@
       // do a status lookup on the box
       // only do this if the box is not selected
       // the server might change the validity for new select/examine
-      imapCommand *cmd =
+      CommandPtr cmd =
         doCommand (imapCommand::clientStatus (aBox, "UIDVALIDITY"));
       completeQueue.removeAll (cmd);
       validity = getStatus ().uidValidity ();
@@ -1991,12 +1990,12 @@
   if (getState() == ISTATE_NO) return;
   if (getState() == ISTATE_SELECT && metaData("expunge") == "auto")
   {
-    imapCommand *cmd = doCommand (imapCommand::clientExpunge());
+    CommandPtr cmd = doCommand (imapCommand::clientExpunge());
     completeQueue.removeAll (cmd);
   }
   if (getState() != ISTATE_CONNECT)
   {
-    imapCommand *cmd = doCommand (imapCommand::clientLogout());
+    CommandPtr cmd = doCommand (imapCommand::clientLogout());
     completeQueue.removeAll (cmd);
   }
   disconnectFromHost();
@@ -2027,14 +2026,14 @@
     myTLS  = metaData("tls");
     kDebug(7116) <<"myAuth:" << myAuth;
 
-    imapCommand *cmd;
+    CommandPtr cmd;
 
     unhandled.clear ();
     if (!alreadyConnected) while (!parseLoop ()) {}   //get greeting
     QString greeting;
     if (!unhandled.isEmpty()) greeting = unhandled.first().trimmed();
     unhandled.clear ();       //get rid of it
-    cmd = doCommand (new imapCommand ("CAPABILITY", ""));
+    cmd = doCommand (CommandPtr(new imapCommand ("CAPABILITY", "")));
 
     kDebug(7116) <<"IMAP4: setHost: capability";
     for (QStringList::const_iterator it = imapCapabilities.constBegin ();
@@ -2043,7 +2042,6 @@
       kDebug(7116) <<"'" << (*it) <<"'";
     }
     completeQueue.removeAll (cmd);
-    delete cmd;
 
     if (!hasCapability("IMAP4") && !hasCapability("IMAP4rev1"))
     {
@@ -2066,14 +2064,14 @@
     if ((myTLS == "on" /*###|| ( canUseTLS() && myTLS != "off")*/) &&
         hasCapability(QString("STARTTLS")))
     {
-      imapCommand *cmd = doCommand (imapCommand::clientStartTLS());
+      CommandPtr cmd = doCommand (imapCommand::clientStartTLS());
       if (cmd->result () == "OK")
       {
         completeQueue.removeAll(cmd);
         if (startSsl())
         {
           kDebug(7116) <<"TLS mode has been enabled.";
-          imapCommand *cmd2 = doCommand (new imapCommand ("CAPABILITY", ""));
+          CommandPtr cmd2 = doCommand (CommandPtr(new imapCommand ("CAPABILITY", "")));
           for (QStringList::const_iterator it = imapCapabilities.constBegin ();
                                      it != imapCapabilities.constEnd (); ++it)
           {
@@ -2084,11 +2082,9 @@
           kWarning(7116) <<"TLS mode setup has failed.  Aborting.";
           error (ERR_COULD_NOT_LOGIN, i18n("Starting TLS failed."));
           closeConnection();
-          delete cmd;
           return false;
         }
       } else completeQueue.removeAll(cmd);
-      delete cmd;
     }
 
     if (!myAuth.isEmpty () && myAuth != "*"
@@ -2148,7 +2144,6 @@
         kDebug(7116) <<"makeLogin - registered namespaces";
       }
       completeQueue.removeAll (cmd);
-      delete cmd;
     }
     // get the default delimiter (empty listing)
     cmd = doCommand( imapCommand::clientList("", "") );
@@ -2168,7 +2163,6 @@
       }
     }
     completeQueue.removeAll (cmd);
-    delete cmd;
   } else {
     kDebug(7116) <<"makeLogin - NO login";
   }
@@ -2420,7 +2414,7 @@
         } else
         {
           // start a listing for the box to get the type
-          imapCommand *cmd;
+          CommandPtr cmd;
 
           cmd = doCommand (imapCommand::clientList ("", _box));
           if (cmd->result () == "OK")
@@ -2595,7 +2589,7 @@
 {
   if (aBox.isEmpty()) return false;
 
-  imapCommand *cmd = 0;
+  CommandPtr cmd;
 
   if (aBox != getCurrentBox () || (!getSelected().readWrite() && !readonly))
   {
Index: kioslave/imap4/imapcommand.cpp
===================================================================
--- kioslave/imap4/imapcommand.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/imap4/imapcommand.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -156,19 +156,19 @@
     return id() + ' ' + command() + ' ' + parameter() + "\r\n";
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientNoop ()
 {
-  return new imapCommand ("NOOP", "");
+  return CommandPtr( new imapCommand ("NOOP", "") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientFetch (ulong uid, const QString & fields, bool nouid)
 {
-  return clientFetch (uid, uid, fields, nouid);
+  return CommandPtr( clientFetch (uid, uid, fields, nouid) );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientFetch (ulong fromUid, ulong toUid, const QString & fields,
                           bool nouid)
 {
@@ -185,173 +185,173 @@
   return clientFetch (uid, fields, nouid);
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientFetch (const QString & sequence, const QString & fields,
                           bool nouid)
 {
-  return new imapCommand (nouid ? "FETCH" : "UID FETCH",
-                          sequence + " (" + fields + ')');
+  return CommandPtr( new imapCommand (nouid ? "FETCH" : "UID FETCH",
+                                      sequence + " (" + fields + ')') );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientList (const QString & reference, const QString & path,
                          bool lsub)
 {
-  return new imapCommand (lsub ? "LSUB" : "LIST",
+  return CommandPtr( new imapCommand (lsub ? "LSUB" : "LIST",
                           QString ("\"") + KIMAP::encodeImapFolderName (reference) +
-                          "\" \"" + KIMAP::encodeImapFolderName (path) + "\"");
+                          "\" \"" + KIMAP::encodeImapFolderName (path) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientSelect (const QString & path, bool examine)
 {
   Q_UNUSED(examine);
   /** @note We use always SELECT, because UW-IMAP doesn't check for new mail, when
      used with the "mbox driver" and the folder is opened with EXAMINE
      and Courier can't append to a mailbox that is in EXAMINE state */
-  return new imapCommand ("SELECT",
-                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"");
+  return CommandPtr( new imapCommand ("SELECT",
+                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientClose()
 {
-  return new imapCommand("CLOSE", "");
+  return CommandPtr( new imapCommand("CLOSE", "") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientCopy (const QString & box, const QString & sequence,
                          bool nouid)
 {
-  return new imapCommand (nouid ? "COPY" : "UID COPY",
-                          sequence + " \"" + KIMAP::encodeImapFolderName (box) + "\"");
+  return CommandPtr( new imapCommand (nouid ? "COPY" : "UID COPY",
+                          sequence + " \"" + KIMAP::encodeImapFolderName (box) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientAppend (const QString & box, const QString & flags,
                            ulong size)
 {
-  return new imapCommand ("APPEND",
+  return CommandPtr( new imapCommand ("APPEND",
                           "\"" + KIMAP::encodeImapFolderName (box) + "\" " +
                           ((flags.isEmpty()) ? "" : ('(' + flags + ") ")) +
-                          '{' + QString::number(size) + '}');
+                          '{' + QString::number(size) + '}') );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientStatus (const QString & path, const QString & parameters)
 {
-  return new imapCommand ("STATUS",
+  return CommandPtr( new imapCommand ("STATUS",
                           QString ("\"") + KIMAP::encodeImapFolderName (path) +
-                          "\" (" + parameters + ")");
+                          "\" (" + parameters + ")") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientCreate (const QString & path)
 {
-  return new imapCommand ("CREATE",
-                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"");
+  return CommandPtr( new imapCommand ("CREATE",
+                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientDelete (const QString & path)
 {
-  return new imapCommand ("DELETE",
-                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"");
+  return CommandPtr( new imapCommand ("DELETE",
+                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientSubscribe (const QString & path)
 {
-  return new imapCommand ("SUBSCRIBE",
-                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"");
+  return CommandPtr( new imapCommand ("SUBSCRIBE",
+                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientUnsubscribe (const QString & path)
 {
-  return new imapCommand ("UNSUBSCRIBE",
-                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"");
+  return CommandPtr(  new imapCommand ("UNSUBSCRIBE",
+                          QString ("\"") + KIMAP::encodeImapFolderName (path) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientExpunge ()
 {
-  return new imapCommand ("EXPUNGE", QString (""));
+  return CommandPtr( new imapCommand ("EXPUNGE", QString ("")) );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientRename (const QString & src, const QString & dest)
 {
-  return new imapCommand ("RENAME",
+  return CommandPtr( new imapCommand ("RENAME",
                           QString ("\"") + KIMAP::encodeImapFolderName (src) +
-                          "\" \"" + KIMAP::encodeImapFolderName (dest) + "\"");
+                          "\" \"" + KIMAP::encodeImapFolderName (dest) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientSearch (const QString & search, bool nouid)
 {
-  return new imapCommand (nouid ? "SEARCH" : "UID SEARCH", search);
+  return CommandPtr( new imapCommand (nouid ? "SEARCH" : "UID SEARCH", search) );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientStore (const QString & set, const QString & item,
                           const QString & data, bool nouid)
 {
-  return new imapCommand (nouid ? "STORE" : "UID STORE",
-                          set + ' ' + item + " (" + data + ')');
+  return CommandPtr( new imapCommand (nouid ? "STORE" : "UID STORE",
+                          set + ' ' + item + " (" + data + ')') );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientLogout ()
 {
-  return new imapCommand ("LOGOUT", "");
+  return CommandPtr( new imapCommand ("LOGOUT", "") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientStartTLS ()
 {
-  return new imapCommand ("STARTTLS", "");
+  return CommandPtr( new imapCommand ("STARTTLS", "") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientSetACL( const QString& box, const QString& user, const QString& acl )
 {
-  return new imapCommand ("SETACL", QString("\"") + KIMAP::encodeImapFolderName (box)
+  return CommandPtr( new imapCommand ("SETACL", QString("\"") + KIMAP::encodeImapFolderName (box)
                           + "\" \"" + KIMAP::encodeImapFolderName (user)
-                          + "\" \"" + KIMAP::encodeImapFolderName (acl) + "\"");
+                          + "\" \"" + KIMAP::encodeImapFolderName (acl) + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientDeleteACL( const QString& box, const QString& user )
 {
-  return new imapCommand ("DELETEACL", QString("\"") + KIMAP::encodeImapFolderName (box)
+  return CommandPtr( new imapCommand ("DELETEACL", QString("\"") + KIMAP::encodeImapFolderName (box)
                           + "\" \"" + KIMAP::encodeImapFolderName (user)
-                          + "\"");
+                          + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientGetACL( const QString& box )
 {
-  return new imapCommand ("GETACL", QString("\"") + KIMAP::encodeImapFolderName (box)
-                          + "\"");
+  return CommandPtr( new imapCommand ("GETACL", QString("\"") + KIMAP::encodeImapFolderName (box)
+                          + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientListRights( const QString& box, const QString& user )
 {
-  return new imapCommand ("LISTRIGHTS", QString("\"") + KIMAP::encodeImapFolderName (box)
+  return CommandPtr( new imapCommand ("LISTRIGHTS", QString("\"") + KIMAP::encodeImapFolderName (box)
                           + "\" \"" + KIMAP::encodeImapFolderName (user)
-                          + "\"");
+                          + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientMyRights( const QString& box )
 {
-  return new imapCommand ("MYRIGHTS", QString("\"") + KIMAP::encodeImapFolderName (box)
-                          + "\"");
+  return CommandPtr( new imapCommand ("MYRIGHTS", QString("\"") + KIMAP::encodeImapFolderName (box)
+                          + "\"") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientSetAnnotation( const QString& box, const QString& entry, const QMap<QString, QString>& attributes )
 {
   QString parameter = QString("\"") + KIMAP::encodeImapFolderName (box)
@@ -367,10 +367,10 @@
   // Turn last space into a ')'
   parameter[parameter.length()-1] = ')';
 
-  return new imapCommand ("SETANNOTATION", parameter);
+  return CommandPtr( new imapCommand ("SETANNOTATION", parameter) );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientGetAnnotation( const QString& box, const QString& entry, const QStringList& attributeNames )
 {
   QString parameter = QString("\"") + KIMAP::encodeImapFolderName (box)
@@ -386,25 +386,25 @@
     // Turn last space into a ')'
     parameter[parameter.length()-1] = ')';
   }
-  return new imapCommand ("GETANNOTATION", parameter);
+  return CommandPtr( new imapCommand ("GETANNOTATION", parameter) );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientNamespace()
 {
-  return new imapCommand("NAMESPACE", "");
+  return CommandPtr( new imapCommand("NAMESPACE", "") );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientGetQuotaroot( const QString& box )
 {
   QString parameter = QString("\"") + KIMAP::encodeImapFolderName (box) + '"';
-  return new imapCommand ("GETQUOTAROOT", parameter);
+  return CommandPtr( new imapCommand ("GETQUOTAROOT", parameter) );
 }
 
-imapCommand *
+CommandPtr
 imapCommand::clientCustom( const QString& command, const QString& arguments )
 {
-  return new imapCommand (command, arguments);
+  return CommandPtr( new imapCommand (command, arguments) );
 }
 
Index: kioslave/imap4/imapcommand.h
===================================================================
--- kioslave/imap4/imapcommand.h	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/imap4/imapcommand.h	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -27,6 +27,11 @@
 #include <QString>
 #include <QMap>
 
+#include <boost/shared_ptr.hpp>
+
+class imapCommand;
+typedef boost::shared_ptr<imapCommand> CommandPtr;
+
 /**
  *  @brief encapulate a IMAP command
  *  @author Svenn Carstens
@@ -143,7 +148,7 @@
    * @brief Create a NOOP command
    * @return a NOOP imapCommand
    */
-  static imapCommand *clientNoop ();
+  static CommandPtr clientNoop ();
   /**
    * @fn static imapCommand *clientFetch (ulong uid, const QString & fields, bool nouid = false);
    * @brief Create a FETCH command
@@ -153,8 +158,8 @@
    * @return a FETCH imapCommand
    * Fetch a single uid
    */
-  static imapCommand *clientFetch (ulong uid, const QString & fields,
-                                   bool nouid = false);
+  static CommandPtr clientFetch (ulong uid, const QString & fields,
+                                 bool nouid = false);
   /**
    * @fn static imapCommand *clientFetch (ulong fromUid, ulong toUid, const QString & fields, bool nouid = false);
    * @brief Create a FETCH command
@@ -165,9 +170,9 @@
    * @return a FETCH imapCommand
    * Fetch a range of uids
    */
-  static imapCommand *clientFetch (ulong fromUid, ulong toUid,
-                                   const QString & fields, bool nouid =
-                                   false);
+  static CommandPtr clientFetch (ulong fromUid, ulong toUid,
+                                 const QString & fields, bool nouid =
+                                 false);
   /**
    * @fn static imapCommand *clientFetch (const QString & sequence, const QString & fields, bool nouid = false);
    * @brief Create a FETCH command
@@ -178,9 +183,9 @@
    * Fetch a range of uids. The other clientFetch functions are just
    * wrappers around this function.
    */
-  static imapCommand *clientFetch (const QString & sequence,
-                                   const QString & fields, bool nouid =
-                                   false);
+  static CommandPtr clientFetch (const QString & sequence,
+                                 const QString & fields, bool nouid =
+                                 false);
   /**
    * @fn static imapCommand *clientList (const QString & reference, const QString & path, bool lsub = false);
    * @brief Create a LIST command
@@ -189,8 +194,8 @@
    * @param lsub Perform a LIST or a LSUB command
    * @return a LIST imapCommand
    */
-  static imapCommand *clientList (const QString & reference,
-                                  const QString & path, bool lsub = false);
+  static CommandPtr clientList (const QString & reference,
+                                const QString & path, bool lsub = false);
   /**
    * @fn static imapCommand *clientSelect (const QString & path, bool examine = false);
    * @brief Create a SELECT command
@@ -198,22 +203,22 @@
    * @param lsub Perform a SELECT or a EXAMINE command
    * @return a SELECT imapCommand
    */
-  static imapCommand *clientSelect (const QString & path, bool examine =
-                                    false);
+  static CommandPtr clientSelect (const QString & path, bool examine =
+                                  false);
   /**
    * @fn static imapCommand *clientClose();
    * @brief Create a CLOSE command
    * @return a CLOSE imapCommand
    */
-  static imapCommand *clientClose();
+  static CommandPtr clientClose();
   /**
    * @brief Create a STATUS command
    * @param path
    * @param parameters
    * @return a STATUS imapCommand
    */
-  static imapCommand *clientStatus (const QString & path,
-                                    const QString & parameters);
+  static CommandPtr clientStatus (const QString & path,
+                                  const QString & parameters);
   /**
    * @brief Create a COPY command
    * @param box
@@ -221,9 +226,9 @@
    * @param nouid Perform a COPY or UID COPY command
    * @return a COPY imapCommand
    */
-  static imapCommand *clientCopy (const QString & box,
-                                  const QString & sequence, bool nouid =
-                                  false);
+  static CommandPtr clientCopy (const QString & box,
+                                const QString & sequence, bool nouid =
+                                false);
   /**
    * @brief Create a APPEND command
    * @param box
@@ -231,53 +236,53 @@
    * @param size
    * @return a APPEND imapCommand
    */
-  static imapCommand *clientAppend (const QString & box,
-                                    const QString & flags, ulong size);
+  static CommandPtr clientAppend (const QString & box,
+                                  const QString & flags, ulong size);
   /**
    * @brief Create a CREATE command
    * @param path
    * @return a CREATE imapCommand
    */
-  static imapCommand *clientCreate (const QString & path);
+  static CommandPtr clientCreate (const QString & path);
   /**
    * @brief Create a DELETE command
    * @param path
    * @return a DELETE imapCommand
    */
-  static imapCommand *clientDelete (const QString & path);
+  static CommandPtr clientDelete (const QString & path);
   /**
    * @brief Create a SUBSCRIBE command
    * @param path
    * @return a SUBSCRIBE imapCommand
    */
-  static imapCommand *clientSubscribe (const QString & path);
+  static CommandPtr clientSubscribe (const QString & path);
   /**
    * @brief Create a UNSUBSCRIBE command
    * @param path
    * @return a UNSUBSCRIBE imapCommand
    */
-  static imapCommand *clientUnsubscribe (const QString & path);
+  static CommandPtr clientUnsubscribe (const QString & path);
   /**
    * @brief Create a EXPUNGE command
    * @return a EXPUNGE imapCommand
    */
-  static imapCommand *clientExpunge ();
+  static CommandPtr clientExpunge ();
   /**
    * @brief Create a RENAME command
    * @param src Source
    * @param dest Destination
    * @return a RENAME imapCommand
    */
-  static imapCommand *clientRename (const QString & src,
-                                    const QString & dest);
+  static CommandPtr clientRename (const QString & src,
+                                  const QString & dest);
   /**
    * @brief Create a SEARCH command
    * @param search
    * @param nouid Perform a UID SEARCH or a SEARCH command
    * @return a SEARCH imapCommand
    */
-  static imapCommand *clientSearch (const QString & search, bool nouid =
-                                    false);
+  static CommandPtr clientSearch (const QString & search, bool nouid =
+                                  false);
   /**
    * @brief Create a STORE command
    * @param set
@@ -286,18 +291,18 @@
    * @param nouid Perform a UID STORE or a STORE command
    * @return a STORE imapCommand
    */
-  static imapCommand *clientStore (const QString & set, const QString & item,
-                                   const QString & data, bool nouid = false);
+  static CommandPtr clientStore (const QString & set, const QString & item,
+                                 const QString & data, bool nouid = false);
   /**
    * @brief Create a LOGOUT command
    * @return a LOGOUT imapCommand
    */
-  static imapCommand *clientLogout ();
+  static CommandPtr clientLogout ();
   /**
    * @brief Create a STARTTLS command
    * @return a STARTTLS imapCommand
    */
-  static imapCommand *clientStartTLS ();
+  static CommandPtr clientStartTLS ();
 
   //////////// ACL support (RFC 2086) /////////////
   /**
@@ -307,7 +312,7 @@
    * @param acl access right modification (starting with optional +/-)
    * @return a SETACL imapCommand
    */
-  static imapCommand *clientSetACL ( const QString& box, const QString& user, const QString& acl );
+  static CommandPtr clientSetACL ( const QString& box, const QString& user, const QString& acl );
 
   /**
    * @brief Create a DELETEACL command
@@ -315,14 +320,14 @@
    * @param user authentication identifier
    * @return a DELETEACL imapCommand
    */
-  static imapCommand *clientDeleteACL ( const QString& box, const QString& user );
+  static CommandPtr clientDeleteACL ( const QString& box, const QString& user );
 
   /**
    * @brief Create a GETACL command
    * @param box mailbox name
    * @return a GETACL imapCommand
    */
-  static imapCommand *clientGetACL ( const QString& box );
+  static CommandPtr clientGetACL ( const QString& box );
 
   /**
    * @brief Create a LISTRIGHTS command
@@ -330,14 +335,14 @@
    * @param user authentication identifier
    * @return a LISTRIGHTS imapCommand
    */
-  static imapCommand *clientListRights ( const QString& box, const QString& user );
+  static CommandPtr clientListRights ( const QString& box, const QString& user );
 
   /**
    * @brief Create a MYRIGHTS command
    * @param box mailbox name
    * @return a MYRIGHTS imapCommand
    */
-  static imapCommand *clientMyRights ( const QString& box );
+  static CommandPtr clientMyRights ( const QString& box );
 
   //////////// ANNOTATEMORE support /////////////
   /**
@@ -347,7 +352,7 @@
    * @param attributes map of attribute names + values
    * @return a SETANNOTATION imapCommand
    */
-  static imapCommand *clientSetAnnotation ( const QString& box, const QString& entry, const QMap<QString, QString>& attributes );
+  static CommandPtr clientSetAnnotation ( const QString& box, const QString& entry, const QMap<QString, QString>& attributes );
 
   /**
    * @brief Create a GETANNOTATION command
@@ -356,20 +361,20 @@
    * @param attributeNames attribute specifier
    * @return a GETANNOTATION imapCommand
    */
-  static imapCommand *clientGetAnnotation ( const QString& box, const QString& entry, const QStringList& attributeNames );
+  static CommandPtr clientGetAnnotation ( const QString& box, const QString& entry, const QStringList& attributeNames );
 
   /**
    * @brief Create a NAMESPACE command
    * @return a NAMESPACE imapCommand
    */
-  static imapCommand *clientNamespace ();
+  static CommandPtr clientNamespace ();
 
   /**
    * @brief Create a GETQUOTAROOT command
    * @param box mailbox name
    * @return a GETQUOTAROOT imapCommand
    */
-  static imapCommand *clientGetQuotaroot ( const QString& box );
+  static CommandPtr clientGetQuotaroot ( const QString& box );
 
   /**
    * @brief Create a custom command
@@ -377,7 +382,7 @@
    * @param arguments The custom arguments
    * @return a custom imapCommand
    */
-  static imapCommand *clientCustom ( const QString& command, const QString& arguments );
+  static CommandPtr clientCustom ( const QString& command, const QString& arguments );
 
 protected:
   QString aCommand;
Index: kioslave/imap4/CMakeLists.txt
===================================================================
--- kioslave/imap4/CMakeLists.txt	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/imap4/CMakeLists.txt	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -1,6 +1,7 @@
 
 set(imap4_optional_includes)
 set(imap4_optional_libs)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${KDE4_ENABLE_EXCEPTIONS}")
 
 if (SASL2_FOUND)
    set(imap4_optional_includes ${imap4_optional_includes} ${SASL2_INCLUDE_DIR})
@@ -12,7 +13,7 @@
 
 configure_file(imap4-config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/imap4-config.h)
 
-include_directories(${imap4_optional_includes})
+include_directories(${imap4_optional_includes} ${Boost_INCLUDE_DIR})
 
 ########### next target ###############
 
Index: kioslave/sieve/sieve.protocol
===================================================================
--- kioslave/sieve/sieve.protocol	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kioslave/sieve/sieve.protocol	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -22,6 +22,7 @@
 Description[fr]=Un module d'entrée / sortie pour le protocole de filtrage de messagerie Sieve
 Description[ga]=ioslave le haghaidh prótacail scagtha ríomhphoist Sieve
 Description[gl]=Un ioslave para o protocolo de filtrado de correo Sieve
+Description[hne]=सीव मेल फिल्टरिंग प्रोटोकाल बर आईओस्लेव
 Description[hu]=KDE-protokoll a Sieve levélszűrő protokollhoz
 Description[it]=Un IOSlave per il protocollo di filtraggio della posta Sieve
 Description[ja]=Sieve メールフィルタリングプロトコルのための ioslave
Index: mailtransport/kcm_mailtransport.desktop
===================================================================
--- mailtransport/kcm_mailtransport.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ mailtransport/kcm_mailtransport.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -23,6 +23,7 @@
 Name[fr]=Transport de courriel
 Name[ga]=Iompar Ríomhphoist
 Name[gl]=Transporte de correo
+Name[hne]=मेल ट्रांसपोर्ट
 Name[hu]=Levéltovábbítás
 Name[it]=Trasporto della posta
 Name[ja]=メール送信手段
Index: kabc/plugins/file/file.desktop
===================================================================
--- kabc/plugins/file/file.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/plugins/file/file.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[fr]=Fichier
 Name[ga]=Comhad
 Name[gl]=Ficheiro
+Name[hne]=फाइल
 Name[hu]=Fájl
 Name[ja]=ファイル
 Name[km]=ឯកសារ
Index: kabc/plugins/ldapkio/ldapkio.desktop
===================================================================
--- kabc/plugins/ldapkio/ldapkio.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/plugins/ldapkio/ldapkio.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Name=LDAP
+Name[hne]=एलडीएपी
 Name[sr]=ЛДАП
 Name[th]=ระบบ LDAP
 Name[x-test]=xxLDAPxx
Index: kabc/plugins/net/resourcenet.cpp
===================================================================
--- kabc/plugins/net/resourcenet.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/plugins/net/resourcenet.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -241,6 +241,7 @@
 
   if ( ok ) {
     saveToFile( &tempFile );
+    tempFile.flush();
   }
 
   if ( !ok ) {
@@ -273,6 +274,7 @@
   bool ok = createLocalTempFile();
   if ( ok ) {
     saveToFile( mTempFile );
+    mTempFile->flush();
   }
 
   if ( !ok ) {
Index: kabc/plugins/net/net.desktop
===================================================================
--- kabc/plugins/net/net.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/plugins/net/net.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[fr]=Réseau
 Name[ga]=Líonra
 Name[gl]=Rede
+Name[hne]=नेटवर्क
 Name[hu]=Hálózat
 Name[it]=Rete
 Name[ja]=ネットワーク
Index: kabc/plugins/dir/dir.desktop
===================================================================
--- kabc/plugins/dir/dir.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/plugins/dir/dir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[fr]=Dossier
 Name[ga]=Comhadlann
 Name[gl]=Cartafol
+Name[hne]=डिरेक्टरी
 Name[hu]=Könyvtár
 Name[it]=Elenco
 Name[ja]=ディレクトリ
Index: kabc/kabc_manager.desktop
===================================================================
--- kabc/kabc_manager.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/kabc_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[et]=Kontaktid
 Name[ga]=Teagmhálacha
 Name[gl]=Contactos
+Name[hne]=सम्पर्क
 Name[hu]=Partnerek
 Name[it]=Contatti
 Name[ja]=連絡先
Index: kabc/addresslineedit.cpp
===================================================================
--- kabc/addresslineedit.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/addresslineedit.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -90,7 +90,7 @@
                         mParent, SLOT( slotCompletion() ) );
 
       KCompletionBox *box = mParent->completionBox();
-      mParent->connect( box, SIGNAL( highlighted( const QString& ) ),
+      mParent->connect( box, SIGNAL( currentTextChanged( const QString& ) ),
                         mParent, SLOT( slotPopupCompletion( const QString& ) ) );
       mParent->connect( box, SIGNAL( userCancelled( const QString& ) ),
                         SLOT( userCancelled( const QString& ) ) );
Index: kabc/formats/binary.desktop
===================================================================
--- kabc/formats/binary.desktop	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ kabc/formats/binary.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -11,6 +11,7 @@
 Name[fr]=Binaire
 Name[ga]=Dénártha
 Name[gl]=Binario
+Name[hne]=द्विचर
 Name[hu]=Bináris
 Name[it]=Binario
 Name[ja]=バイナリ
Index: akonadi/selftestdialog.cpp
===================================================================
--- akonadi/selftestdialog.cpp	(.../tags/KDE/4.2.1/kdepimlibs)	(wersja 942069)
+++ akonadi/selftestdialog.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 942069)
@@ -541,6 +541,9 @@
 void SelfTestDialog::saveReport()
 {
   const QString fileName =  KFileDialog::getSaveFileName( KUrl(), QString(), this, i18n("Save Test Report") );
+  if ( fileName.isEmpty() )
+    return;
+
   QFile file( fileName );
   if ( !file.open( QFile::ReadWrite ) ) {
     KMessageBox::error( this, i18n( "Could not open file '%1'", fileName ) );
