Index: kresources/kresources_manager.desktop
===================================================================
--- kresources/kresources_manager.desktop	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kresources/kresources_manager.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -27,7 +27,7 @@
 Name[pt]=Gestor do KResources
 Name[pt_BR]=Gerenciador do KResource
 Name[ro]=Gestionar KResurse
-Name[ru]=Управление источниками
+Name[ru]=Управление источниками данных
 Name[se]=KResources-gieđahalli
 Name[sl]=Upravnik KResources
 Name[sr]=Менаџер к‑ресурса
@@ -69,7 +69,7 @@
 Comment[pt]=Gestor do KResources
 Comment[pt_BR]=Gerenciador do KResource
 Comment[ro]=Gestionar KResurse
-Comment[ru]=Управление источниками
+Comment[ru]=Управление источниками данных
 Comment[se]=KResources-gieđahalli
 Comment[sl]=Upravnik KResources
 Comment[sr]=Менаџер к‑ресурса
Index: kresources/kresources.desktop
===================================================================
--- kresources/kresources.desktop	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kresources/kresources.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -36,7 +36,7 @@
 Name[pt]=Recursos do KDE
 Name[pt_BR]=Fontes de dados do KDE
 Name[ro]=Resurse KDE
-Name[ru]=Источники KDE
+Name[ru]=Источники данных
 Name[se]=KDE-resurssat
 Name[sl]=Viri KDE
 Name[sr]=КДЕ ресурси
@@ -77,7 +77,7 @@
 Comment[pt]=Configurar os Recursos do KDE
 Comment[pt_BR]=Configurar as fontes de dados do KDE
 Comment[ro]=Configurare resurse KDE
-Comment[ru]=Настроить источники KDE
+Comment[ru]=Настроить источники данных KDE
 Comment[se]=Heivet KDE-resurssaid
 Comment[sl]=Nastavi vire KDE
 Comment[sr]=Подеси ресурсе КДЕ‑а
Index: kcal/localdir.desktop
===================================================================
--- kcal/localdir.desktop	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kcal/localdir.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -69,6 +69,7 @@
 Comment[sr]=Пружа приступ календарским ставкама, складиштеним у појединачним фајловима у датој фасцикли.
 Comment[sr@latin]=Pruža pristup kalendarskim stavkama, skladištenim u pojedinačnim fajlovima u datoj fascikli.
 Comment[sv]=Ger tillgång till kalenderobjekt, vart och ett lagrat i en enskild lokal fil, i en given katalog
+Comment[tr]=Belirli bir dizinde, her biri tek bir dosyada depolanmış takvim ögelerine erişim sağlar
 Comment[uk]=Надає доступ до записів календаря, кожен з яких зберігається у окремому файлі у вказаному каталозі
 Comment[x-test]=xxProvides access to calendar items, each stored in a single file, in a given directoryxx
 Comment[zh_CN]=提供对被存储在给定目录下单独文件中的日历内容的访问支持
Index: kcal/local.desktop
===================================================================
--- kcal/local.desktop	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kcal/local.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -69,6 +69,7 @@
 Comment[sr]=Пружа приступ календару складиштеном у једном локалном фајлу
 Comment[sr@latin]=Pruža pristup kalendaru skladištenom u jednom lokalnom fajlu
 Comment[sv]=Ger tillgång till en kalender lagrad i en enda lokal fil
+Comment[tr]=TEk bir yerel dosyada depolanmış bir takvime erişim sağlar
 Comment[uk]=Надає доступ до календаря, що зберігається у окремому локальному файлі
 Comment[x-test]=xxProvides access to a calendar stored in a single local filexx
 Comment[zh_CN]=提供对被存储在单独的本地文件中的日历的访问支持
Index: kcal/incidenceformatter.cpp
===================================================================
--- kcal/incidenceformatter.cpp	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kcal/incidenceformatter.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -1476,8 +1476,7 @@
       html += "</td></tr><tr>";
     }
     html += "<td>";
-
-    if ( !existingIncidence ) {
+    if ( incidence && !existingIncidence ) {
       // Accept
       html += helper->makeLink( "accept", i18nc( "accept to-do request", "[Accept]" ) );
       html += "</td><td> &nbsp; </td><td>";
Index: kabc/addresseedialog.cpp
===================================================================
--- kabc/addresseedialog.cpp	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kabc/addresseedialog.cpp	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -317,7 +317,7 @@
 
 void AddresseeDialog::Private::removeSelected()
 {
-  QList<QTreeWidgetItem*> selected = mAddresseeList->selectedItems();
+  QList<QTreeWidgetItem*> selected = mSelectedList->selectedItems();
   if ( selected.count() == 0 ) {
     return;
   }
Index: kabc/plugins/ldapkio/ldapkio.desktop
===================================================================
--- kabc/plugins/ldapkio/ldapkio.desktop	(.../tags/KDE/4.2.2/kdepimlibs)	(wersja 952918)
+++ kabc/plugins/ldapkio/ldapkio.desktop	(.../branches/KDE/4.2/kdepimlibs)	(wersja 952918)
@@ -31,6 +31,7 @@
 Comment[sr]=Пружа приступ контактима складиштеним на серверу ЛДАП именика
 Comment[sr@latin]=Pruža pristup kontaktima skladištenim na serveru LDAP imenika
 Comment[sv]=Ger tillgång till kontakter lagrade på en LDAP-katalogserver
+Comment[tr]=Bir LDAP dizin sunucusunda depolanmış kişilere erişim sağlar
 Comment[uk]=Надає доступ до контактів, які зберігаються на сервері каталогів LDAP
 Comment[x-test]=xxProvides access to contacts stored on a LDAP directory serverxx
 Comment[zh_CN]=提供对被存储在 LDAP 目录服务器上的联系人的访问支持
